{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Example: Mothur MiSeq SOP\n",
    "\n",
    "This example usage of rhea uses the Mothur MiSeq SOP."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This example requires the following packages to run (that are not a requirement of rhea itself):\n",
    "\n",
    "* wget\n",
    "\n",
    "These can be installed at the command line with `pip install <package_name>`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "\n",
    "# change working directory to allow import of package without installation\n",
    "work_dir = os.getcwd()\n",
    "base_dir = os.path.join(work_dir, '..', '..')\n",
    "os.chdir(base_dir)\n",
    "\n",
    "from rhea import Mothur\n",
    "os.chdir(work_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The rhea package revolves around the central `Mothur` class. Before running any mothur commands we need to create an instance of the `Mothur` class that we will use to run our `mothur` commands. We can also set attributes on this `Mothur` class to apply different configuration options that affect the way that `mothur` will be executed, and how the output will be displayed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# create Mothur class instance\n",
    "mothur = Mothur()\n",
    "\n",
    "# configure the mothur object\n",
    "mothur.suppress_logfile = True\n",
    "mothur.verbosity = 1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First we need to download all the files required for running this example."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Example data already exists. Skipping download and extraction.\n"
     ]
    }
   ],
   "source": [
    "import wget\n",
    "from zipfile import ZipFile\n",
    "\n",
    "# download and unpack example data\n",
    "if not os.path.isdir('MiSeq_SOP'):  \n",
    "    \n",
    "    # onyl redownload data if neither the packed or unpacked forma already exist\n",
    "    if not os.path.isfile('MiSeqSOPData.zip'):\n",
    "        print('Downloading Example Data...')\n",
    "        data_archive = wget.download('https://www.mothur.org/w/images/d/d6/MiSeqSOPData.zip')\n",
    "\n",
    "    # unpack data\n",
    "    print('\\nExtracting example data...')\n",
    "    data = ZipFile(data_archive)\n",
    "    data.extractall()\n",
    "    print('done!')\n",
    "    \n",
    "else:\n",
    "    print('Example data already exists. Skipping download and extraction.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SILVA reference alignment already exists. Skipping download and extraction.\n"
     ]
    }
   ],
   "source": [
    "# download and unpack SILVA reference alignment\n",
    "if not os.path.isdir('silva.bacteria'):  \n",
    "    \n",
    "    # onyl redownload data if neither the packed or unpacked forma already exist\n",
    "    if not os.path.isfile('Silva.bacteria.zip'):\n",
    "        print('Downloading SILVA referece alignment...')\n",
    "        data_archive = wget.download('https://www.mothur.org/w/images/9/98/Silva.bacteria.zip')\n",
    "\n",
    "    # unpack data\n",
    "    print('\\nExtracting SILVA reference alignment...')\n",
    "    data = ZipFile(data_archive)\n",
    "    data.extractall()\n",
    "    print('done!')\n",
    "    \n",
    "else:\n",
    "    print('SILVA reference alignment already exists. Skipping download and extraction.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RDP training set already exists. Skipping download and extraction.\n"
     ]
    }
   ],
   "source": [
    "# download and unpack SILVA reference alignment\n",
    "if not os.path.isfile('trainset9_032012.pds.fasta') or not os.path.isfile('trainset9_032012.pds.tax'):  \n",
    "    \n",
    "    # onyl redownload data if neither the packed or unpacked forma already exist\n",
    "    if not os.path.isfile('Trainset9_032012.pds.zip'):\n",
    "        print('Downloading RDP training set...')\n",
    "        data_archive = wget.download('https://www.mothur.org/w/images/5/59/Trainset9_032012.pds.zip')\n",
    "        \n",
    "    # unpack data\n",
    "    print('\\nExtracting RDP training set...')\n",
    "    data = ZipFile(data_archive)\n",
    "    data.extractall()\n",
    "    print('done!')\n",
    "    \n",
    "else:\n",
    "    print('RDP training set already exists. Skipping download and extraction.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "## Not doing this right now as make.contigs is broken, using the one provided in the example data instead\n",
    "# mothur.make.file(inputdir='MiSeq_SOP', type='fastq', prefix='stability')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > set.dir(input=MiSeq_SOP,output=.)\n",
      "Mothur's directories:\n",
      "outputDir=C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\\n",
      "inputDir=MiSeq_SOP\\\n",
      "\n",
      "mothur > make.contigs(file=stability.files)\n",
      "\n",
      "Using 1 processors.\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D0_S188_L001_R1_001.fastq - MiSeq_SOP\\F3D0_S188_L001_R2_001.fastq (files 1 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "6000\n",
      "7000\n",
      "7793\n",
      "Done.\n",
      "\n",
      "It took 5 secs to assemble 7793 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D141_S207_L001_R1_001.fastq - MiSeq_SOP\\F3D141_S207_L001_R2_001.fastq (files 2 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "5958\n",
      "Done.\n",
      "\n",
      "It took 4 secs to assemble 5958 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D142_S208_L001_R1_001.fastq - MiSeq_SOP\\F3D142_S208_L001_R2_001.fastq (files 3 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "3183\n",
      "Done.\n",
      "\n",
      "It took 2 secs to assemble 3183 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D143_S209_L001_R1_001.fastq - MiSeq_SOP\\F3D143_S209_L001_R2_001.fastq (files 4 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "3178\n",
      "Done.\n",
      "\n",
      "It took 3 secs to assemble 3178 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D144_S210_L001_R1_001.fastq - MiSeq_SOP\\F3D144_S210_L001_R2_001.fastq (files 5 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "4827\n",
      "Done.\n",
      "\n",
      "It took 3 secs to assemble 4827 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D145_S211_L001_R1_001.fastq - MiSeq_SOP\\F3D145_S211_L001_R2_001.fastq (files 6 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "6000\n",
      "7000\n",
      "7377\n",
      "Done.\n",
      "\n",
      "It took 5 secs to assemble 7377 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D146_S212_L001_R1_001.fastq - MiSeq_SOP\\F3D146_S212_L001_R2_001.fastq (files 7 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "5021\n",
      "Done.\n",
      "\n",
      "It took 4 secs to assemble 5021 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D147_S213_L001_R1_001.fastq - MiSeq_SOP\\F3D147_S213_L001_R2_001.fastq (files 8 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "6000\n",
      "7000\n",
      "8000\n",
      "9000\n",
      "10000\n",
      "11000\n",
      "12000\n",
      "13000\n",
      "14000\n",
      "15000\n",
      "16000\n",
      "17000\n",
      "17070\n",
      "Done.\n",
      "\n",
      "It took 11 secs to assemble 17070 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D148_S214_L001_R1_001.fastq - MiSeq_SOP\\F3D148_S214_L001_R2_001.fastq (files 9 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "6000\n",
      "7000\n",
      "8000\n",
      "9000\n",
      "10000\n",
      "11000\n",
      "12000\n",
      "12405\n",
      "Done.\n",
      "\n",
      "It took 9 secs to assemble 12405 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D149_S215_L001_R1_001.fastq - MiSeq_SOP\\F3D149_S215_L001_R2_001.fastq (files 10 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "6000\n",
      "7000\n",
      "8000\n",
      "9000\n",
      "10000\n",
      "11000\n",
      "12000\n",
      "13000\n",
      "13083\n",
      "Done.\n",
      "\n",
      "It took 9 secs to assemble 13083 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D150_S216_L001_R1_001.fastq - MiSeq_SOP\\F3D150_S216_L001_R2_001.fastq (files 11 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "5509\n",
      "Done.\n",
      "\n",
      "It took 4 secs to assemble 5509 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D1_S189_L001_R1_001.fastq - MiSeq_SOP\\F3D1_S189_L001_R2_001.fastq (files 12 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "5869\n",
      "Done.\n",
      "\n",
      "It took 4 secs to assemble 5869 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D2_S190_L001_R1_001.fastq - MiSeq_SOP\\F3D2_S190_L001_R2_001.fastq (files 13 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "6000\n",
      "7000\n",
      "8000\n",
      "9000\n",
      "10000\n",
      "11000\n",
      "12000\n",
      "13000\n",
      "14000\n",
      "15000\n",
      "16000\n",
      "17000\n",
      "18000\n",
      "19000\n",
      "19620\n",
      "Done.\n",
      "\n",
      "It took 13 secs to assemble 19620 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D3_S191_L001_R1_001.fastq - MiSeq_SOP\\F3D3_S191_L001_R2_001.fastq (files 14 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "6000\n",
      "6758\n",
      "Done.\n",
      "\n",
      "It took 5 secs to assemble 6758 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D5_S193_L001_R1_001.fastq - MiSeq_SOP\\F3D5_S193_L001_R2_001.fastq (files 15 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "4448\n",
      "Done.\n",
      "\n",
      "It took 3 secs to assemble 4448 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D6_S194_L001_R1_001.fastq - MiSeq_SOP\\F3D6_S194_L001_R2_001.fastq (files 16 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "6000\n",
      "7000\n",
      "7989\n",
      "Done.\n",
      "\n",
      "It took 5 secs to assemble 7989 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D7_S195_L001_R1_001.fastq - MiSeq_SOP\\F3D7_S195_L001_R2_001.fastq (files 17 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "5129\n",
      "Done.\n",
      "\n",
      "It took 4 secs to assemble 5129 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D8_S196_L001_R1_001.fastq - MiSeq_SOP\\F3D8_S196_L001_R2_001.fastq (files 18 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "5294\n",
      "Done.\n",
      "\n",
      "It took 3 secs to assemble 5294 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\F3D9_S197_L001_R1_001.fastq - MiSeq_SOP\\F3D9_S197_L001_R2_001.fastq (files 19 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "6000\n",
      "7000\n",
      "7070\n",
      "Done.\n",
      "\n",
      "It took 5 secs to assemble 7070 reads.\n",
      "\n",
      "\n",
      ">>>>>\tProcessing file pair MiSeq_SOP\\Mock_S280_L001_R1_001.fastq - MiSeq_SOP\\Mock_S280_L001_R2_001.fastq (files 20 of 20)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "4779\n",
      "Done.\n",
      "\n",
      "It took 4 secs to assemble 4779 reads.\n",
      "\n",
      "It took 105 secs to process 152360 sequences.\n",
      "\n",
      "Group count: \n",
      "F3D0\t7793\n",
      "F3D1\t5869\n",
      "F3D141\t5958\n",
      "F3D142\t3183\n",
      "F3D143\t3178\n",
      "F3D144\t4827\n",
      "F3D145\t7377\n",
      "F3D146\t5021\n",
      "F3D147\t17070\n",
      "F3D148\t12405\n",
      "F3D149\t13083\n",
      "F3D150\t5509\n",
      "F3D2\t19620\n",
      "F3D3\t6758\n",
      "F3D5\t4448\n",
      "F3D6\t7989\n",
      "F3D7\t5129\n",
      "F3D8\t5294\n",
      "F3D9\t7070\n",
      "Mock\t4779\n",
      "\n",
      "Total of all groups is 152360\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.fasta\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.qual\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.contigs.report\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.scrap.contigs.fasta\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.scrap.contigs.qual\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.contigs.groups\n",
      "\n",
      "[WARNING]: your sequence names contained ':'.  I changed them to '_' to avoid problems in your downstream analysis.\n",
      "\n",
      "mothur > get.current()\n",
      "\n",
      "Current RAM usage: 6.95547 Gigabytes. Total Ram: 63.8537 Gigabytes.\n",
      "\n",
      "Current files saved by mothur:\n",
      "fasta=C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.fasta\n",
      "group=C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.contigs.groups\n",
      "qfile=C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.qual\n",
      "contigsreport=C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.contigs.report\n",
      "\n",
      "Current input directory saved by mothur: MiSeq_SOP\\\n",
      "\n",
      "Current output directory saved by mothur: C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\\n",
      "\n",
      "Current default directory saved by mothur: C:\\Users\\campen\\mothur\\1.39.5\\mothur\\mothur\\\n",
      "\n",
      "Current working directory: C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\current_files.summary\n",
      "\n",
      "\n",
      "mothur > quit()\n",
      "\n",
      "\n",
      "<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<^>>>>>>>>>>>>>>>>>>>>>>>>>>>>>\n",
      "<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<^>>>>>>>>>>>>>>>>>>>>>>>>>>>>>\n",
      "<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<^>>>>>>>>>>>>>>>>>>>>>>>>>>>>>\n",
      "Detected 1 [WARNING] messages, please review.\n",
      "<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<^>>>>>>>>>>>>>>>>>>>>>>>>>>>>>\n",
      "<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<^>>>>>>>>>>>>>>>>>>>>>>>>>>>>>\n",
      "<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<^>>>>>>>>>>>>>>>>>>>>>>>>>>>>>\n",
      "mothur > set.dir(input=C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\)\n",
      "Mothur's directories:\n",
      "inputDir=C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# we're setting up the input and output directories explicitly to prevent anything weird\n",
    "mothur = mothur.set.dir(input='MiSeq_SOP', output='.')\n",
    "mothur = mothur.make.contigs(file='stability.files')\n",
    "\n",
    "# resetting the input directory after reading in the sequencing files from the sub-folder\n",
    "mothur = mothur.set.dir(input=mothur.current_dirs['output'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > summary.seqs()\n",
      "Using C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.fasta as input file for the fasta parameter.\n",
      "\n",
      "Using 1 processors.\n",
      "\n",
      "\t\tStart\tEnd\tNBases\tAmbigs\tPolymer\tNumSeqs\n",
      "Minimum:\t1\t248\t248\t0\t3\t1\n",
      "2.5%-tile:\t1\t252\t252\t0\t3\t3810\n",
      "25%-tile:\t1\t252\t252\t0\t4\t38091\n",
      "Median: \t1\t252\t252\t0\t4\t76181\n",
      "75%-tile:\t1\t253\t253\t0\t5\t114271\n",
      "97.5%-tile:\t1\t253\t253\t6\t6\t148552\n",
      "Maximum:\t1\t502\t502\t249\t243\t152360\n",
      "Mean:\t1\t252.811\t252.811\t0.70063\t4.44854\n",
      "# of Seqs:\t152360\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.summary\n",
      "\n",
      "It took 1 secs to summarize 152360 sequences.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.summary.seqs()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > screen.seqs(fasta=current,group=current,maxambig=0,maxlength=275)\n",
      "Using C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.fasta as input file for the fasta parameter.\n",
      "Using C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.contigs.groups as input file for the group parameter.\n",
      "\n",
      "Using 1 processors.\n",
      "Processing sequence: 100\n",
      "Processing sequence: 200\n",
      "Processing sequence: 300\n",
      "Processing sequence: 400\n",
      "Processing sequence: 500\n",
      "Processing sequence: 600\n",
      "Processing sequence: 700\n",
      "Processing sequence: 800\n",
      "Processing sequence: 900\n",
      "Processing sequence: 1000\n",
      "Processing sequence: 1100\n",
      "Processing sequence: 1200\n",
      "Processing sequence: 1300\n",
      "Processing sequence: 1400\n",
      "Processing sequence: 1500\n",
      "Processing sequence: 1600\n",
      "Processing sequence: 1700\n",
      "Processing sequence: 1800\n",
      "Processing sequence: 1900\n",
      "Processing sequence: 2000\n",
      "Processing sequence: 2100\n",
      "Processing sequence: 2200\n",
      "Processing sequence: 2300\n",
      "Processing sequence: 2400\n",
      "Processing sequence: 2500\n",
      "Processing sequence: 2600\n",
      "Processing sequence: 2700\n",
      "Processing sequence: 2800\n",
      "Processing sequence: 2900\n",
      "Processing sequence: 3000\n",
      "Processing sequence: 3100\n",
      "Processing sequence: 3200\n",
      "Processing sequence: 3300\n",
      "Processing sequence: 3400\n",
      "Processing sequence: 3500\n",
      "Processing sequence: 3600\n",
      "Processing sequence: 3700\n",
      "Processing sequence: 3800\n",
      "Processing sequence: 3900\n",
      "Processing sequence: 4000\n",
      "Processing sequence: 4100\n",
      "Processing sequence: 4200\n",
      "Processing sequence: 4300\n",
      "Processing sequence: 4400\n",
      "Processing sequence: 4500\n",
      "Processing sequence: 4600\n",
      "Processing sequence: 4700\n",
      "Processing sequence: 4800\n",
      "Processing sequence: 4900\n",
      "Processing sequence: 5000\n",
      "Processing sequence: 5100\n",
      "Processing sequence: 5200\n",
      "Processing sequence: 5300\n",
      "Processing sequence: 5400\n",
      "Processing sequence: 5500\n",
      "Processing sequence: 5600\n",
      "Processing sequence: 5700\n",
      "Processing sequence: 5800\n",
      "Processing sequence: 5900\n",
      "Processing sequence: 6000\n",
      "Processing sequence: 6100\n",
      "Processing sequence: 6200\n",
      "Processing sequence: 6300\n",
      "Processing sequence: 6400\n",
      "Processing sequence: 6500\n",
      "Processing sequence: 6600\n",
      "Processing sequence: 6700\n",
      "Processing sequence: 6800\n",
      "Processing sequence: 6900\n",
      "Processing sequence: 7000\n",
      "Processing sequence: 7100\n",
      "Processing sequence: 7200\n",
      "Processing sequence: 7300\n",
      "Processing sequence: 7400\n",
      "Processing sequence: 7500\n",
      "Processing sequence: 7600\n",
      "Processing sequence: 7700\n",
      "Processing sequence: 7800\n",
      "Processing sequence: 7900\n",
      "Processing sequence: 8000\n",
      "Processing sequence: 8100\n",
      "Processing sequence: 8200\n",
      "Processing sequence: 8300\n",
      "Processing sequence: 8400\n",
      "Processing sequence: 8500\n",
      "Processing sequence: 8600\n",
      "Processing sequence: 8700\n",
      "Processing sequence: 8800\n",
      "Processing sequence: 8900\n",
      "Processing sequence: 9000\n",
      "Processing sequence: 9100\n",
      "Processing sequence: 9200\n",
      "Processing sequence: 9300\n",
      "Processing sequence: 9400\n",
      "Processing sequence: 9500\n",
      "Processing sequence: 9600\n",
      "Processing sequence: 9700\n",
      "Processing sequence: 9800\n",
      "Processing sequence: 9900\n",
      "Processing sequence: 10000\n",
      "Processing sequence: 10100\n",
      "Processing sequence: 10200\n",
      "Processing sequence: 10300\n",
      "Processing sequence: 10400\n",
      "Processing sequence: 10500\n",
      "Processing sequence: 10600\n",
      "Processing sequence: 10700\n",
      "Processing sequence: 10800\n",
      "Processing sequence: 10900\n",
      "Processing sequence: 11000\n",
      "Processing sequence: 11100\n",
      "Processing sequence: 11200\n",
      "Processing sequence: 11300\n",
      "Processing sequence: 11400\n",
      "Processing sequence: 11500\n",
      "Processing sequence: 11600\n",
      "Processing sequence: 11700\n",
      "Processing sequence: 11800\n",
      "Processing sequence: 11900\n",
      "Processing sequence: 12000\n",
      "Processing sequence: 12100\n",
      "Processing sequence: 12200\n",
      "Processing sequence: 12300\n",
      "Processing sequence: 12400\n",
      "Processing sequence: 12500\n",
      "Processing sequence: 12600\n",
      "Processing sequence: 12700\n",
      "Processing sequence: 12800\n",
      "Processing sequence: 12900\n",
      "Processing sequence: 13000\n",
      "Processing sequence: 13100\n",
      "Processing sequence: 13200\n",
      "Processing sequence: 13300\n",
      "Processing sequence: 13400\n",
      "Processing sequence: 13500\n",
      "Processing sequence: 13600\n",
      "Processing sequence: 13700\n",
      "Processing sequence: 13800\n",
      "Processing sequence: 13900\n",
      "Processing sequence: 14000\n",
      "Processing sequence: 14100\n",
      "Processing sequence: 14200\n",
      "Processing sequence: 14300\n",
      "Processing sequence: 14400\n",
      "Processing sequence: 14500\n",
      "Processing sequence: 14600\n",
      "Processing sequence: 14700\n",
      "Processing sequence: 14800\n",
      "Processing sequence: 14900\n",
      "Processing sequence: 15000\n",
      "Processing sequence: 15100\n",
      "Processing sequence: 15200\n",
      "Processing sequence: 15300\n",
      "Processing sequence: 15400\n",
      "Processing sequence: 15500\n",
      "Processing sequence: 15600\n",
      "Processing sequence: 15700\n",
      "Processing sequence: 15800\n",
      "Processing sequence: 15900\n",
      "Processing sequence: 16000\n",
      "Processing sequence: 16100\n",
      "Processing sequence: 16200\n",
      "Processing sequence: 16300\n",
      "Processing sequence: 16400\n",
      "Processing sequence: 16500\n",
      "Processing sequence: 16600\n",
      "Processing sequence: 16700\n",
      "Processing sequence: 16800\n",
      "Processing sequence: 16900\n",
      "Processing sequence: 17000\n",
      "Processing sequence: 17100\n",
      "Processing sequence: 17200\n",
      "Processing sequence: 17300\n",
      "Processing sequence: 17400\n",
      "Processing sequence: 17500\n",
      "Processing sequence: 17600\n",
      "Processing sequence: 17700\n",
      "Processing sequence: 17800\n",
      "Processing sequence: 17900\n",
      "Processing sequence: 18000\n",
      "Processing sequence: 18100\n",
      "Processing sequence: 18200\n",
      "Processing sequence: 18300\n",
      "Processing sequence: 18400\n",
      "Processing sequence: 18500\n",
      "Processing sequence: 18600\n",
      "Processing sequence: 18700\n",
      "Processing sequence: 18800\n",
      "Processing sequence: 18900\n",
      "Processing sequence: 19000\n",
      "Processing sequence: 19100\n",
      "Processing sequence: 19200\n",
      "Processing sequence: 19300\n",
      "Processing sequence: 19400\n",
      "Processing sequence: 19500\n",
      "Processing sequence: 19600\n",
      "Processing sequence: 19700\n",
      "Processing sequence: 19800\n",
      "Processing sequence: 19900\n",
      "Processing sequence: 20000\n",
      "Processing sequence: 20100\n",
      "Processing sequence: 20200\n",
      "Processing sequence: 20300\n",
      "Processing sequence: 20400\n",
      "Processing sequence: 20500\n",
      "Processing sequence: 20600\n",
      "Processing sequence: 20700\n",
      "Processing sequence: 20800\n",
      "Processing sequence: 20900\n",
      "Processing sequence: 21000\n",
      "Processing sequence: 21100\n",
      "Processing sequence: 21200\n",
      "Processing sequence: 21300\n",
      "Processing sequence: 21400\n",
      "Processing sequence: 21500\n",
      "Processing sequence: 21600\n",
      "Processing sequence: 21700\n",
      "Processing sequence: 21800\n",
      "Processing sequence: 21900\n",
      "Processing sequence: 22000\n",
      "Processing sequence: 22100\n",
      "Processing sequence: 22200\n",
      "Processing sequence: 22300\n",
      "Processing sequence: 22400\n",
      "Processing sequence: 22500\n",
      "Processing sequence: 22600\n",
      "Processing sequence: 22700\n",
      "Processing sequence: 22800\n",
      "Processing sequence: 22900\n",
      "Processing sequence: 23000\n",
      "Processing sequence: 23100\n",
      "Processing sequence: 23200\n",
      "Processing sequence: 23300\n",
      "Processing sequence: 23400\n",
      "Processing sequence: 23500\n",
      "Processing sequence: 23600\n",
      "Processing sequence: 23700\n",
      "Processing sequence: 23800\n",
      "Processing sequence: 23900\n",
      "Processing sequence: 24000\n",
      "Processing sequence: 24100\n",
      "Processing sequence: 24200\n",
      "Processing sequence: 24300\n",
      "Processing sequence: 24400\n",
      "Processing sequence: 24500\n",
      "Processing sequence: 24600\n",
      "Processing sequence: 24700\n",
      "Processing sequence: 24800\n",
      "Processing sequence: 24900\n",
      "Processing sequence: 25000\n",
      "Processing sequence: 25100\n",
      "Processing sequence: 25200\n",
      "Processing sequence: 25300\n",
      "Processing sequence: 25400\n",
      "Processing sequence: 25500\n",
      "Processing sequence: 25600\n",
      "Processing sequence: 25700\n",
      "Processing sequence: 25800\n",
      "Processing sequence: 25900\n",
      "Processing sequence: 26000\n",
      "Processing sequence: 26100\n",
      "Processing sequence: 26200\n",
      "Processing sequence: 26300\n",
      "Processing sequence: 26400\n",
      "Processing sequence: 26500\n",
      "Processing sequence: 26600\n",
      "Processing sequence: 26700\n",
      "Processing sequence: 26800\n",
      "Processing sequence: 26900\n",
      "Processing sequence: 27000\n",
      "Processing sequence: 27100\n",
      "Processing sequence: 27200\n",
      "Processing sequence: 27300\n",
      "Processing sequence: 27400\n",
      "Processing sequence: 27500\n",
      "Processing sequence: 27600\n",
      "Processing sequence: 27700\n",
      "Processing sequence: 27800\n",
      "Processing sequence: 27900\n",
      "Processing sequence: 28000\n",
      "Processing sequence: 28100\n",
      "Processing sequence: 28200\n",
      "Processing sequence: 28300\n",
      "Processing sequence: 28400\n",
      "Processing sequence: 28500\n",
      "Processing sequence: 28600\n",
      "Processing sequence: 28700\n",
      "Processing sequence: 28800\n",
      "Processing sequence: 28900\n",
      "Processing sequence: 29000\n",
      "Processing sequence: 29100\n",
      "Processing sequence: 29200\n",
      "Processing sequence: 29300\n",
      "Processing sequence: 29400\n",
      "Processing sequence: 29500\n",
      "Processing sequence: 29600\n",
      "Processing sequence: 29700\n",
      "Processing sequence: 29800\n",
      "Processing sequence: 29900\n",
      "Processing sequence: 30000\n",
      "Processing sequence: 30100\n",
      "Processing sequence: 30200\n",
      "Processing sequence: 30300\n",
      "Processing sequence: 30400\n",
      "Processing sequence: 30500\n",
      "Processing sequence: 30600\n",
      "Processing sequence: 30700\n",
      "Processing sequence: 30800\n",
      "Processing sequence: 30900\n",
      "Processing sequence: 31000\n",
      "Processing sequence: 31100\n",
      "Processing sequence: 31200\n",
      "Processing sequence: 31300\n",
      "Processing sequence: 31400\n",
      "Processing sequence: 31500\n",
      "Processing sequence: 31600\n",
      "Processing sequence: 31700\n",
      "Processing sequence: 31800\n",
      "Processing sequence: 31900\n",
      "Processing sequence: 32000\n",
      "Processing sequence: 32100\n",
      "Processing sequence: 32200\n",
      "Processing sequence: 32300\n",
      "Processing sequence: 32400\n",
      "Processing sequence: 32500\n",
      "Processing sequence: 32600\n",
      "Processing sequence: 32700\n",
      "Processing sequence: 32800\n",
      "Processing sequence: 32900\n",
      "Processing sequence: 33000\n",
      "Processing sequence: 33100\n",
      "Processing sequence: 33200\n",
      "Processing sequence: 33300\n",
      "Processing sequence: 33400\n",
      "Processing sequence: 33500\n",
      "Processing sequence: 33600\n",
      "Processing sequence: 33700\n",
      "Processing sequence: 33800\n",
      "Processing sequence: 33900\n",
      "Processing sequence: 34000\n",
      "Processing sequence: 34100\n",
      "Processing sequence: 34200\n",
      "Processing sequence: 34300\n",
      "Processing sequence: 34400\n",
      "Processing sequence: 34500\n",
      "Processing sequence: 34600\n",
      "Processing sequence: 34700\n",
      "Processing sequence: 34800\n",
      "Processing sequence: 34900\n",
      "Processing sequence: 35000\n",
      "Processing sequence: 35100\n",
      "Processing sequence: 35200\n",
      "Processing sequence: 35300\n",
      "Processing sequence: 35400\n",
      "Processing sequence: 35500\n",
      "Processing sequence: 35600\n",
      "Processing sequence: 35700\n",
      "Processing sequence: 35800\n",
      "Processing sequence: 35900\n",
      "Processing sequence: 36000\n",
      "Processing sequence: 36100\n",
      "Processing sequence: 36200\n",
      "Processing sequence: 36300\n",
      "Processing sequence: 36400\n",
      "Processing sequence: 36500\n",
      "Processing sequence: 36600\n",
      "Processing sequence: 36700\n",
      "Processing sequence: 36800\n",
      "Processing sequence: 36900\n",
      "Processing sequence: 37000\n",
      "Processing sequence: 37100\n",
      "Processing sequence: 37200\n",
      "Processing sequence: 37300\n",
      "Processing sequence: 37400\n",
      "Processing sequence: 37500\n",
      "Processing sequence: 37600\n",
      "Processing sequence: 37700\n",
      "Processing sequence: 37800\n",
      "Processing sequence: 37900\n",
      "Processing sequence: 38000\n",
      "Processing sequence: 38100\n",
      "Processing sequence: 38200\n",
      "Processing sequence: 38300\n",
      "Processing sequence: 38400\n",
      "Processing sequence: 38500\n",
      "Processing sequence: 38600\n",
      "Processing sequence: 38700\n",
      "Processing sequence: 38800\n",
      "Processing sequence: 38900\n",
      "Processing sequence: 39000\n",
      "Processing sequence: 39100\n",
      "Processing sequence: 39200\n",
      "Processing sequence: 39300\n",
      "Processing sequence: 39400\n",
      "Processing sequence: 39500\n",
      "Processing sequence: 39600\n",
      "Processing sequence: 39700\n",
      "Processing sequence: 39800\n",
      "Processing sequence: 39900\n",
      "Processing sequence: 40000\n",
      "Processing sequence: 40100\n",
      "Processing sequence: 40200\n",
      "Processing sequence: 40300\n",
      "Processing sequence: 40400\n",
      "Processing sequence: 40500\n",
      "Processing sequence: 40600\n",
      "Processing sequence: 40700\n",
      "Processing sequence: 40800\n",
      "Processing sequence: 40900\n",
      "Processing sequence: 41000\n",
      "Processing sequence: 41100\n",
      "Processing sequence: 41200\n",
      "Processing sequence: 41300\n",
      "Processing sequence: 41400\n",
      "Processing sequence: 41500\n",
      "Processing sequence: 41600\n",
      "Processing sequence: 41700\n",
      "Processing sequence: 41800\n",
      "Processing sequence: 41900\n",
      "Processing sequence: 42000\n",
      "Processing sequence: 42100\n",
      "Processing sequence: 42200\n",
      "Processing sequence: 42300\n",
      "Processing sequence: 42400\n",
      "Processing sequence: 42500\n",
      "Processing sequence: 42600\n",
      "Processing sequence: 42700\n",
      "Processing sequence: 42800\n",
      "Processing sequence: 42900\n",
      "Processing sequence: 43000\n",
      "Processing sequence: 43100\n",
      "Processing sequence: 43200\n",
      "Processing sequence: 43300\n",
      "Processing sequence: 43400\n",
      "Processing sequence: 43500\n",
      "Processing sequence: 43600\n",
      "Processing sequence: 43700\n",
      "Processing sequence: 43800\n",
      "Processing sequence: 43900\n",
      "Processing sequence: 44000\n",
      "Processing sequence: 44100\n",
      "Processing sequence: 44200\n",
      "Processing sequence: 44300\n",
      "Processing sequence: 44400\n",
      "Processing sequence: 44500\n",
      "Processing sequence: 44600\n",
      "Processing sequence: 44700\n",
      "Processing sequence: 44800\n",
      "Processing sequence: 44900\n",
      "Processing sequence: 45000\n",
      "Processing sequence: 45100\n",
      "Processing sequence: 45200\n",
      "Processing sequence: 45300\n",
      "Processing sequence: 45400\n",
      "Processing sequence: 45500\n",
      "Processing sequence: 45600\n",
      "Processing sequence: 45700\n",
      "Processing sequence: 45800\n",
      "Processing sequence: 45900\n",
      "Processing sequence: 46000\n",
      "Processing sequence: 46100\n",
      "Processing sequence: 46200\n",
      "Processing sequence: 46300\n",
      "Processing sequence: 46400\n",
      "Processing sequence: 46500\n",
      "Processing sequence: 46600\n",
      "Processing sequence: 46700\n",
      "Processing sequence: 46800\n",
      "Processing sequence: 46900\n",
      "Processing sequence: 47000\n",
      "Processing sequence: 47100\n",
      "Processing sequence: 47200\n",
      "Processing sequence: 47300\n",
      "Processing sequence: 47400\n",
      "Processing sequence: 47500\n",
      "Processing sequence: 47600\n",
      "Processing sequence: 47700\n",
      "Processing sequence: 47800\n",
      "Processing sequence: 47900\n",
      "Processing sequence: 48000\n",
      "Processing sequence: 48100\n",
      "Processing sequence: 48200\n",
      "Processing sequence: 48300\n",
      "Processing sequence: 48400\n",
      "Processing sequence: 48500\n",
      "Processing sequence: 48600\n",
      "Processing sequence: 48700\n",
      "Processing sequence: 48800\n",
      "Processing sequence: 48900\n",
      "Processing sequence: 49000\n",
      "Processing sequence: 49100\n",
      "Processing sequence: 49200\n",
      "Processing sequence: 49300\n",
      "Processing sequence: 49400\n",
      "Processing sequence: 49500\n",
      "Processing sequence: 49600\n",
      "Processing sequence: 49700\n",
      "Processing sequence: 49800\n",
      "Processing sequence: 49900\n",
      "Processing sequence: 50000\n",
      "Processing sequence: 50100\n",
      "Processing sequence: 50200\n",
      "Processing sequence: 50300\n",
      "Processing sequence: 50400\n",
      "Processing sequence: 50500\n",
      "Processing sequence: 50600\n",
      "Processing sequence: 50700\n",
      "Processing sequence: 50800\n",
      "Processing sequence: 50900\n",
      "Processing sequence: 51000\n",
      "Processing sequence: 51100\n",
      "Processing sequence: 51200\n",
      "Processing sequence: 51300\n",
      "Processing sequence: 51400\n",
      "Processing sequence: 51500\n",
      "Processing sequence: 51600\n",
      "Processing sequence: 51700\n",
      "Processing sequence: 51800\n",
      "Processing sequence: 51900\n",
      "Processing sequence: 52000\n",
      "Processing sequence: 52100\n",
      "Processing sequence: 52200\n",
      "Processing sequence: 52300\n",
      "Processing sequence: 52400\n",
      "Processing sequence: 52500\n",
      "Processing sequence: 52600\n",
      "Processing sequence: 52700\n",
      "Processing sequence: 52800\n",
      "Processing sequence: 52900\n",
      "Processing sequence: 53000\n",
      "Processing sequence: 53100\n",
      "Processing sequence: 53200\n",
      "Processing sequence: 53300\n",
      "Processing sequence: 53400\n",
      "Processing sequence: 53500\n",
      "Processing sequence: 53600\n",
      "Processing sequence: 53700\n",
      "Processing sequence: 53800\n",
      "Processing sequence: 53900\n",
      "Processing sequence: 54000\n",
      "Processing sequence: 54100\n",
      "Processing sequence: 54200\n",
      "Processing sequence: 54300\n",
      "Processing sequence: 54400\n",
      "Processing sequence: 54500\n",
      "Processing sequence: 54600\n",
      "Processing sequence: 54700\n",
      "Processing sequence: 54800\n",
      "Processing sequence: 54900\n",
      "Processing sequence: 55000\n",
      "Processing sequence: 55100\n",
      "Processing sequence: 55200\n",
      "Processing sequence: 55300\n",
      "Processing sequence: 55400\n",
      "Processing sequence: 55500\n",
      "Processing sequence: 55600\n",
      "Processing sequence: 55700\n",
      "Processing sequence: 55800\n",
      "Processing sequence: 55900\n",
      "Processing sequence: 56000\n",
      "Processing sequence: 56100\n",
      "Processing sequence: 56200\n",
      "Processing sequence: 56300\n",
      "Processing sequence: 56400\n",
      "Processing sequence: 56500\n",
      "Processing sequence: 56600\n",
      "Processing sequence: 56700\n",
      "Processing sequence: 56800\n",
      "Processing sequence: 56900\n",
      "Processing sequence: 57000\n",
      "Processing sequence: 57100\n",
      "Processing sequence: 57200\n",
      "Processing sequence: 57300\n",
      "Processing sequence: 57400\n",
      "Processing sequence: 57500\n",
      "Processing sequence: 57600\n",
      "Processing sequence: 57700\n",
      "Processing sequence: 57800\n",
      "Processing sequence: 57900\n",
      "Processing sequence: 58000\n",
      "Processing sequence: 58100\n",
      "Processing sequence: 58200\n",
      "Processing sequence: 58300\n",
      "Processing sequence: 58400\n",
      "Processing sequence: 58500\n",
      "Processing sequence: 58600\n",
      "Processing sequence: 58700\n",
      "Processing sequence: 58800\n",
      "Processing sequence: 58900\n",
      "Processing sequence: 59000\n",
      "Processing sequence: 59100\n",
      "Processing sequence: 59200\n",
      "Processing sequence: 59300\n",
      "Processing sequence: 59400\n",
      "Processing sequence: 59500\n",
      "Processing sequence: 59600\n",
      "Processing sequence: 59700\n",
      "Processing sequence: 59800\n",
      "Processing sequence: 59900\n",
      "Processing sequence: 60000\n",
      "Processing sequence: 60100\n",
      "Processing sequence: 60200\n",
      "Processing sequence: 60300\n",
      "Processing sequence: 60400\n",
      "Processing sequence: 60500\n",
      "Processing sequence: 60600\n",
      "Processing sequence: 60700\n",
      "Processing sequence: 60800\n",
      "Processing sequence: 60900\n",
      "Processing sequence: 61000\n",
      "Processing sequence: 61100\n",
      "Processing sequence: 61200\n",
      "Processing sequence: 61300\n",
      "Processing sequence: 61400\n",
      "Processing sequence: 61500\n",
      "Processing sequence: 61600\n",
      "Processing sequence: 61700\n",
      "Processing sequence: 61800\n",
      "Processing sequence: 61900\n",
      "Processing sequence: 62000\n",
      "Processing sequence: 62100\n",
      "Processing sequence: 62200\n",
      "Processing sequence: 62300\n",
      "Processing sequence: 62400\n",
      "Processing sequence: 62500\n",
      "Processing sequence: 62600\n",
      "Processing sequence: 62700\n",
      "Processing sequence: 62800\n",
      "Processing sequence: 62900\n",
      "Processing sequence: 63000\n",
      "Processing sequence: 63100\n",
      "Processing sequence: 63200\n",
      "Processing sequence: 63300\n",
      "Processing sequence: 63400\n",
      "Processing sequence: 63500\n",
      "Processing sequence: 63600\n",
      "Processing sequence: 63700\n",
      "Processing sequence: 63800\n",
      "Processing sequence: 63900\n",
      "Processing sequence: 64000\n",
      "Processing sequence: 64100\n",
      "Processing sequence: 64200\n",
      "Processing sequence: 64300\n",
      "Processing sequence: 64400\n",
      "Processing sequence: 64500\n",
      "Processing sequence: 64600\n",
      "Processing sequence: 64700\n",
      "Processing sequence: 64800\n",
      "Processing sequence: 64900\n",
      "Processing sequence: 65000\n",
      "Processing sequence: 65100\n",
      "Processing sequence: 65200\n",
      "Processing sequence: 65300\n",
      "Processing sequence: 65400\n",
      "Processing sequence: 65500\n",
      "Processing sequence: 65600\n",
      "Processing sequence: 65700\n",
      "Processing sequence: 65800\n",
      "Processing sequence: 65900\n",
      "Processing sequence: 66000\n",
      "Processing sequence: 66100\n",
      "Processing sequence: 66200\n",
      "Processing sequence: 66300\n",
      "Processing sequence: 66400\n",
      "Processing sequence: 66500\n",
      "Processing sequence: 66600\n",
      "Processing sequence: 66700\n",
      "Processing sequence: 66800\n",
      "Processing sequence: 66900\n",
      "Processing sequence: 67000\n",
      "Processing sequence: 67100\n",
      "Processing sequence: 67200\n",
      "Processing sequence: 67300\n",
      "Processing sequence: 67400\n",
      "Processing sequence: 67500\n",
      "Processing sequence: 67600\n",
      "Processing sequence: 67700\n",
      "Processing sequence: 67800\n",
      "Processing sequence: 67900\n",
      "Processing sequence: 68000\n",
      "Processing sequence: 68100\n",
      "Processing sequence: 68200\n",
      "Processing sequence: 68300\n",
      "Processing sequence: 68400\n",
      "Processing sequence: 68500\n",
      "Processing sequence: 68600\n",
      "Processing sequence: 68700\n",
      "Processing sequence: 68800\n",
      "Processing sequence: 68900\n",
      "Processing sequence: 69000\n",
      "Processing sequence: 69100\n",
      "Processing sequence: 69200\n",
      "Processing sequence: 69300\n",
      "Processing sequence: 69400\n",
      "Processing sequence: 69500\n",
      "Processing sequence: 69600\n",
      "Processing sequence: 69700\n",
      "Processing sequence: 69800\n",
      "Processing sequence: 69900\n",
      "Processing sequence: 70000\n",
      "Processing sequence: 70100\n",
      "Processing sequence: 70200\n",
      "Processing sequence: 70300\n",
      "Processing sequence: 70400\n",
      "Processing sequence: 70500\n",
      "Processing sequence: 70600\n",
      "Processing sequence: 70700\n",
      "Processing sequence: 70800\n",
      "Processing sequence: 70900\n",
      "Processing sequence: 71000\n",
      "Processing sequence: 71100\n",
      "Processing sequence: 71200\n",
      "Processing sequence: 71300\n",
      "Processing sequence: 71400\n",
      "Processing sequence: 71500\n",
      "Processing sequence: 71600\n",
      "Processing sequence: 71700\n",
      "Processing sequence: 71800\n",
      "Processing sequence: 71900\n",
      "Processing sequence: 72000\n",
      "Processing sequence: 72100\n",
      "Processing sequence: 72200\n",
      "Processing sequence: 72300\n",
      "Processing sequence: 72400\n",
      "Processing sequence: 72500\n",
      "Processing sequence: 72600\n",
      "Processing sequence: 72700\n",
      "Processing sequence: 72800\n",
      "Processing sequence: 72900\n",
      "Processing sequence: 73000\n",
      "Processing sequence: 73100\n",
      "Processing sequence: 73200\n",
      "Processing sequence: 73300\n",
      "Processing sequence: 73400\n",
      "Processing sequence: 73500\n",
      "Processing sequence: 73600\n",
      "Processing sequence: 73700\n",
      "Processing sequence: 73800\n",
      "Processing sequence: 73900\n",
      "Processing sequence: 74000\n",
      "Processing sequence: 74100\n",
      "Processing sequence: 74200\n",
      "Processing sequence: 74300\n",
      "Processing sequence: 74400\n",
      "Processing sequence: 74500\n",
      "Processing sequence: 74600\n",
      "Processing sequence: 74700\n",
      "Processing sequence: 74800\n",
      "Processing sequence: 74900\n",
      "Processing sequence: 75000\n",
      "Processing sequence: 75100\n",
      "Processing sequence: 75200\n",
      "Processing sequence: 75300\n",
      "Processing sequence: 75400\n",
      "Processing sequence: 75500\n",
      "Processing sequence: 75600\n",
      "Processing sequence: 75700\n",
      "Processing sequence: 75800\n",
      "Processing sequence: 75900\n",
      "Processing sequence: 76000\n",
      "Processing sequence: 76100\n",
      "Processing sequence: 76200\n",
      "Processing sequence: 76300\n",
      "Processing sequence: 76400\n",
      "Processing sequence: 76500\n",
      "Processing sequence: 76600\n",
      "Processing sequence: 76700\n",
      "Processing sequence: 76800\n",
      "Processing sequence: 76900\n",
      "Processing sequence: 77000\n",
      "Processing sequence: 77100\n",
      "Processing sequence: 77200\n",
      "Processing sequence: 77300\n",
      "Processing sequence: 77400\n",
      "Processing sequence: 77500\n",
      "Processing sequence: 77600\n",
      "Processing sequence: 77700\n",
      "Processing sequence: 77800\n",
      "Processing sequence: 77900\n",
      "Processing sequence: 78000\n",
      "Processing sequence: 78100\n",
      "Processing sequence: 78200\n",
      "Processing sequence: 78300\n",
      "Processing sequence: 78400\n",
      "Processing sequence: 78500\n",
      "Processing sequence: 78600\n",
      "Processing sequence: 78700\n",
      "Processing sequence: 78800\n",
      "Processing sequence: 78900\n",
      "Processing sequence: 79000\n",
      "Processing sequence: 79100\n",
      "Processing sequence: 79200\n",
      "Processing sequence: 79300\n",
      "Processing sequence: 79400\n",
      "Processing sequence: 79500\n",
      "Processing sequence: 79600\n",
      "Processing sequence: 79700\n",
      "Processing sequence: 79800\n",
      "Processing sequence: 79900\n",
      "Processing sequence: 80000\n",
      "Processing sequence: 80100\n",
      "Processing sequence: 80200\n",
      "Processing sequence: 80300\n",
      "Processing sequence: 80400\n",
      "Processing sequence: 80500\n",
      "Processing sequence: 80600\n",
      "Processing sequence: 80700\n",
      "Processing sequence: 80800\n",
      "Processing sequence: 80900\n",
      "Processing sequence: 81000\n",
      "Processing sequence: 81100\n",
      "Processing sequence: 81200\n",
      "Processing sequence: 81300\n",
      "Processing sequence: 81400\n",
      "Processing sequence: 81500\n",
      "Processing sequence: 81600\n",
      "Processing sequence: 81700\n",
      "Processing sequence: 81800\n",
      "Processing sequence: 81900\n",
      "Processing sequence: 82000\n",
      "Processing sequence: 82100\n",
      "Processing sequence: 82200\n",
      "Processing sequence: 82300\n",
      "Processing sequence: 82400\n",
      "Processing sequence: 82500\n",
      "Processing sequence: 82600\n",
      "Processing sequence: 82700\n",
      "Processing sequence: 82800\n",
      "Processing sequence: 82900\n",
      "Processing sequence: 83000\n",
      "Processing sequence: 83100\n",
      "Processing sequence: 83200\n",
      "Processing sequence: 83300\n",
      "Processing sequence: 83400\n",
      "Processing sequence: 83500\n",
      "Processing sequence: 83600\n",
      "Processing sequence: 83700\n",
      "Processing sequence: 83800\n",
      "Processing sequence: 83900\n",
      "Processing sequence: 84000\n",
      "Processing sequence: 84100\n",
      "Processing sequence: 84200\n",
      "Processing sequence: 84300\n",
      "Processing sequence: 84400\n",
      "Processing sequence: 84500\n",
      "Processing sequence: 84600\n",
      "Processing sequence: 84700\n",
      "Processing sequence: 84800\n",
      "Processing sequence: 84900\n",
      "Processing sequence: 85000\n",
      "Processing sequence: 85100\n",
      "Processing sequence: 85200\n",
      "Processing sequence: 85300\n",
      "Processing sequence: 85400\n",
      "Processing sequence: 85500\n",
      "Processing sequence: 85600\n",
      "Processing sequence: 85700\n",
      "Processing sequence: 85800\n",
      "Processing sequence: 85900\n",
      "Processing sequence: 86000\n",
      "Processing sequence: 86100\n",
      "Processing sequence: 86200\n",
      "Processing sequence: 86300\n",
      "Processing sequence: 86400\n",
      "Processing sequence: 86500\n",
      "Processing sequence: 86600\n",
      "Processing sequence: 86700\n",
      "Processing sequence: 86800\n",
      "Processing sequence: 86900\n",
      "Processing sequence: 87000\n",
      "Processing sequence: 87100\n",
      "Processing sequence: 87200\n",
      "Processing sequence: 87300\n",
      "Processing sequence: 87400\n",
      "Processing sequence: 87500\n",
      "Processing sequence: 87600\n",
      "Processing sequence: 87700\n",
      "Processing sequence: 87800\n",
      "Processing sequence: 87900\n",
      "Processing sequence: 88000\n",
      "Processing sequence: 88100\n",
      "Processing sequence: 88200\n",
      "Processing sequence: 88300\n",
      "Processing sequence: 88400\n",
      "Processing sequence: 88500\n",
      "Processing sequence: 88600\n",
      "Processing sequence: 88700\n",
      "Processing sequence: 88800\n",
      "Processing sequence: 88900\n",
      "Processing sequence: 89000\n",
      "Processing sequence: 89100\n",
      "Processing sequence: 89200\n",
      "Processing sequence: 89300\n",
      "Processing sequence: 89400\n",
      "Processing sequence: 89500\n",
      "Processing sequence: 89600\n",
      "Processing sequence: 89700\n",
      "Processing sequence: 89800\n",
      "Processing sequence: 89900\n",
      "Processing sequence: 90000\n",
      "Processing sequence: 90100\n",
      "Processing sequence: 90200\n",
      "Processing sequence: 90300\n",
      "Processing sequence: 90400\n",
      "Processing sequence: 90500\n",
      "Processing sequence: 90600\n",
      "Processing sequence: 90700\n",
      "Processing sequence: 90800\n",
      "Processing sequence: 90900\n",
      "Processing sequence: 91000\n",
      "Processing sequence: 91100\n",
      "Processing sequence: 91200\n",
      "Processing sequence: 91300\n",
      "Processing sequence: 91400\n",
      "Processing sequence: 91500\n",
      "Processing sequence: 91600\n",
      "Processing sequence: 91700\n",
      "Processing sequence: 91800\n",
      "Processing sequence: 91900\n",
      "Processing sequence: 92000\n",
      "Processing sequence: 92100\n",
      "Processing sequence: 92200\n",
      "Processing sequence: 92300\n",
      "Processing sequence: 92400\n",
      "Processing sequence: 92500\n",
      "Processing sequence: 92600\n",
      "Processing sequence: 92700\n",
      "Processing sequence: 92800\n",
      "Processing sequence: 92900\n",
      "Processing sequence: 93000\n",
      "Processing sequence: 93100\n",
      "Processing sequence: 93200\n",
      "Processing sequence: 93300\n",
      "Processing sequence: 93400\n",
      "Processing sequence: 93500\n",
      "Processing sequence: 93600\n",
      "Processing sequence: 93700\n",
      "Processing sequence: 93800\n",
      "Processing sequence: 93900\n",
      "Processing sequence: 94000\n",
      "Processing sequence: 94100\n",
      "Processing sequence: 94200\n",
      "Processing sequence: 94300\n",
      "Processing sequence: 94400\n",
      "Processing sequence: 94500\n",
      "Processing sequence: 94600\n",
      "Processing sequence: 94700\n",
      "Processing sequence: 94800\n",
      "Processing sequence: 94900\n",
      "Processing sequence: 95000\n",
      "Processing sequence: 95100\n",
      "Processing sequence: 95200\n",
      "Processing sequence: 95300\n",
      "Processing sequence: 95400\n",
      "Processing sequence: 95500\n",
      "Processing sequence: 95600\n",
      "Processing sequence: 95700\n",
      "Processing sequence: 95800\n",
      "Processing sequence: 95900\n",
      "Processing sequence: 96000\n",
      "Processing sequence: 96100\n",
      "Processing sequence: 96200\n",
      "Processing sequence: 96300\n",
      "Processing sequence: 96400\n",
      "Processing sequence: 96500\n",
      "Processing sequence: 96600\n",
      "Processing sequence: 96700\n",
      "Processing sequence: 96800\n",
      "Processing sequence: 96900\n",
      "Processing sequence: 97000\n",
      "Processing sequence: 97100\n",
      "Processing sequence: 97200\n",
      "Processing sequence: 97300\n",
      "Processing sequence: 97400\n",
      "Processing sequence: 97500\n",
      "Processing sequence: 97600\n",
      "Processing sequence: 97700\n",
      "Processing sequence: 97800\n",
      "Processing sequence: 97900\n",
      "Processing sequence: 98000\n",
      "Processing sequence: 98100\n",
      "Processing sequence: 98200\n",
      "Processing sequence: 98300\n",
      "Processing sequence: 98400\n",
      "Processing sequence: 98500\n",
      "Processing sequence: 98600\n",
      "Processing sequence: 98700\n",
      "Processing sequence: 98800\n",
      "Processing sequence: 98900\n",
      "Processing sequence: 99000\n",
      "Processing sequence: 99100\n",
      "Processing sequence: 99200\n",
      "Processing sequence: 99300\n",
      "Processing sequence: 99400\n",
      "Processing sequence: 99500\n",
      "Processing sequence: 99600\n",
      "Processing sequence: 99700\n",
      "Processing sequence: 99800\n",
      "Processing sequence: 99900\n",
      "Processing sequence: 100000\n",
      "Processing sequence: 100100\n",
      "Processing sequence: 100200\n",
      "Processing sequence: 100300\n",
      "Processing sequence: 100400\n",
      "Processing sequence: 100500\n",
      "Processing sequence: 100600\n",
      "Processing sequence: 100700\n",
      "Processing sequence: 100800\n",
      "Processing sequence: 100900\n",
      "Processing sequence: 101000\n",
      "Processing sequence: 101100\n",
      "Processing sequence: 101200\n",
      "Processing sequence: 101300\n",
      "Processing sequence: 101400\n",
      "Processing sequence: 101500\n",
      "Processing sequence: 101600\n",
      "Processing sequence: 101700\n",
      "Processing sequence: 101800\n",
      "Processing sequence: 101900\n",
      "Processing sequence: 102000\n",
      "Processing sequence: 102100\n",
      "Processing sequence: 102200\n",
      "Processing sequence: 102300\n",
      "Processing sequence: 102400\n",
      "Processing sequence: 102500\n",
      "Processing sequence: 102600\n",
      "Processing sequence: 102700\n",
      "Processing sequence: 102800\n",
      "Processing sequence: 102900\n",
      "Processing sequence: 103000\n",
      "Processing sequence: 103100\n",
      "Processing sequence: 103200\n",
      "Processing sequence: 103300\n",
      "Processing sequence: 103400\n",
      "Processing sequence: 103500\n",
      "Processing sequence: 103600\n",
      "Processing sequence: 103700\n",
      "Processing sequence: 103800\n",
      "Processing sequence: 103900\n",
      "Processing sequence: 104000\n",
      "Processing sequence: 104100\n",
      "Processing sequence: 104200\n",
      "Processing sequence: 104300\n",
      "Processing sequence: 104400\n",
      "Processing sequence: 104500\n",
      "Processing sequence: 104600\n",
      "Processing sequence: 104700\n",
      "Processing sequence: 104800\n",
      "Processing sequence: 104900\n",
      "Processing sequence: 105000\n",
      "Processing sequence: 105100\n",
      "Processing sequence: 105200\n",
      "Processing sequence: 105300\n",
      "Processing sequence: 105400\n",
      "Processing sequence: 105500\n",
      "Processing sequence: 105600\n",
      "Processing sequence: 105700\n",
      "Processing sequence: 105800\n",
      "Processing sequence: 105900\n",
      "Processing sequence: 106000\n",
      "Processing sequence: 106100\n",
      "Processing sequence: 106200\n",
      "Processing sequence: 106300\n",
      "Processing sequence: 106400\n",
      "Processing sequence: 106500\n",
      "Processing sequence: 106600\n",
      "Processing sequence: 106700\n",
      "Processing sequence: 106800\n",
      "Processing sequence: 106900\n",
      "Processing sequence: 107000\n",
      "Processing sequence: 107100\n",
      "Processing sequence: 107200\n",
      "Processing sequence: 107300\n",
      "Processing sequence: 107400\n",
      "Processing sequence: 107500\n",
      "Processing sequence: 107600\n",
      "Processing sequence: 107700\n",
      "Processing sequence: 107800\n",
      "Processing sequence: 107900\n",
      "Processing sequence: 108000\n",
      "Processing sequence: 108100\n",
      "Processing sequence: 108200\n",
      "Processing sequence: 108300\n",
      "Processing sequence: 108400\n",
      "Processing sequence: 108500\n",
      "Processing sequence: 108600\n",
      "Processing sequence: 108700\n",
      "Processing sequence: 108800\n",
      "Processing sequence: 108900\n",
      "Processing sequence: 109000\n",
      "Processing sequence: 109100\n",
      "Processing sequence: 109200\n",
      "Processing sequence: 109300\n",
      "Processing sequence: 109400\n",
      "Processing sequence: 109500\n",
      "Processing sequence: 109600\n",
      "Processing sequence: 109700\n",
      "Processing sequence: 109800\n",
      "Processing sequence: 109900\n",
      "Processing sequence: 110000\n",
      "Processing sequence: 110100\n",
      "Processing sequence: 110200\n",
      "Processing sequence: 110300\n",
      "Processing sequence: 110400\n",
      "Processing sequence: 110500\n",
      "Processing sequence: 110600\n",
      "Processing sequence: 110700\n",
      "Processing sequence: 110800\n",
      "Processing sequence: 110900\n",
      "Processing sequence: 111000\n",
      "Processing sequence: 111100\n",
      "Processing sequence: 111200\n",
      "Processing sequence: 111300\n",
      "Processing sequence: 111400\n",
      "Processing sequence: 111500\n",
      "Processing sequence: 111600\n",
      "Processing sequence: 111700\n",
      "Processing sequence: 111800\n",
      "Processing sequence: 111900\n",
      "Processing sequence: 112000\n",
      "Processing sequence: 112100\n",
      "Processing sequence: 112200\n",
      "Processing sequence: 112300\n",
      "Processing sequence: 112400\n",
      "Processing sequence: 112500\n",
      "Processing sequence: 112600\n",
      "Processing sequence: 112700\n",
      "Processing sequence: 112800\n",
      "Processing sequence: 112900\n",
      "Processing sequence: 113000\n",
      "Processing sequence: 113100\n",
      "Processing sequence: 113200\n",
      "Processing sequence: 113300\n",
      "Processing sequence: 113400\n",
      "Processing sequence: 113500\n",
      "Processing sequence: 113600\n",
      "Processing sequence: 113700\n",
      "Processing sequence: 113800\n",
      "Processing sequence: 113900\n",
      "Processing sequence: 114000\n",
      "Processing sequence: 114100\n",
      "Processing sequence: 114200\n",
      "Processing sequence: 114300\n",
      "Processing sequence: 114400\n",
      "Processing sequence: 114500\n",
      "Processing sequence: 114600\n",
      "Processing sequence: 114700\n",
      "Processing sequence: 114800\n",
      "Processing sequence: 114900\n",
      "Processing sequence: 115000\n",
      "Processing sequence: 115100\n",
      "Processing sequence: 115200\n",
      "Processing sequence: 115300\n",
      "Processing sequence: 115400\n",
      "Processing sequence: 115500\n",
      "Processing sequence: 115600\n",
      "Processing sequence: 115700\n",
      "Processing sequence: 115800\n",
      "Processing sequence: 115900\n",
      "Processing sequence: 116000\n",
      "Processing sequence: 116100\n",
      "Processing sequence: 116200\n",
      "Processing sequence: 116300\n",
      "Processing sequence: 116400\n",
      "Processing sequence: 116500\n",
      "Processing sequence: 116600\n",
      "Processing sequence: 116700\n",
      "Processing sequence: 116800\n",
      "Processing sequence: 116900\n",
      "Processing sequence: 117000\n",
      "Processing sequence: 117100\n",
      "Processing sequence: 117200\n",
      "Processing sequence: 117300\n",
      "Processing sequence: 117400\n",
      "Processing sequence: 117500\n",
      "Processing sequence: 117600\n",
      "Processing sequence: 117700\n",
      "Processing sequence: 117800\n",
      "Processing sequence: 117900\n",
      "Processing sequence: 118000\n",
      "Processing sequence: 118100\n",
      "Processing sequence: 118200\n",
      "Processing sequence: 118300\n",
      "Processing sequence: 118400\n",
      "Processing sequence: 118500\n",
      "Processing sequence: 118600\n",
      "Processing sequence: 118700\n",
      "Processing sequence: 118800\n",
      "Processing sequence: 118900\n",
      "Processing sequence: 119000\n",
      "Processing sequence: 119100\n",
      "Processing sequence: 119200\n",
      "Processing sequence: 119300\n",
      "Processing sequence: 119400\n",
      "Processing sequence: 119500\n",
      "Processing sequence: 119600\n",
      "Processing sequence: 119700\n",
      "Processing sequence: 119800\n",
      "Processing sequence: 119900\n",
      "Processing sequence: 120000\n",
      "Processing sequence: 120100\n",
      "Processing sequence: 120200\n",
      "Processing sequence: 120300\n",
      "Processing sequence: 120400\n",
      "Processing sequence: 120500\n",
      "Processing sequence: 120600\n",
      "Processing sequence: 120700\n",
      "Processing sequence: 120800\n",
      "Processing sequence: 120900\n",
      "Processing sequence: 121000\n",
      "Processing sequence: 121100\n",
      "Processing sequence: 121200\n",
      "Processing sequence: 121300\n",
      "Processing sequence: 121400\n",
      "Processing sequence: 121500\n",
      "Processing sequence: 121600\n",
      "Processing sequence: 121700\n",
      "Processing sequence: 121800\n",
      "Processing sequence: 121900\n",
      "Processing sequence: 122000\n",
      "Processing sequence: 122100\n",
      "Processing sequence: 122200\n",
      "Processing sequence: 122300\n",
      "Processing sequence: 122400\n",
      "Processing sequence: 122500\n",
      "Processing sequence: 122600\n",
      "Processing sequence: 122700\n",
      "Processing sequence: 122800\n",
      "Processing sequence: 122900\n",
      "Processing sequence: 123000\n",
      "Processing sequence: 123100\n",
      "Processing sequence: 123200\n",
      "Processing sequence: 123300\n",
      "Processing sequence: 123400\n",
      "Processing sequence: 123500\n",
      "Processing sequence: 123600\n",
      "Processing sequence: 123700\n",
      "Processing sequence: 123800\n",
      "Processing sequence: 123900\n",
      "Processing sequence: 124000\n",
      "Processing sequence: 124100\n",
      "Processing sequence: 124200\n",
      "Processing sequence: 124300\n",
      "Processing sequence: 124400\n",
      "Processing sequence: 124500\n",
      "Processing sequence: 124600\n",
      "Processing sequence: 124700\n",
      "Processing sequence: 124800\n",
      "Processing sequence: 124900\n",
      "Processing sequence: 125000\n",
      "Processing sequence: 125100\n",
      "Processing sequence: 125200\n",
      "Processing sequence: 125300\n",
      "Processing sequence: 125400\n",
      "Processing sequence: 125500\n",
      "Processing sequence: 125600\n",
      "Processing sequence: 125700\n",
      "Processing sequence: 125800\n",
      "Processing sequence: 125900\n",
      "Processing sequence: 126000\n",
      "Processing sequence: 126100\n",
      "Processing sequence: 126200\n",
      "Processing sequence: 126300\n",
      "Processing sequence: 126400\n",
      "Processing sequence: 126500\n",
      "Processing sequence: 126600\n",
      "Processing sequence: 126700\n",
      "Processing sequence: 126800\n",
      "Processing sequence: 126900\n",
      "Processing sequence: 127000\n",
      "Processing sequence: 127100\n",
      "Processing sequence: 127200\n",
      "Processing sequence: 127300\n",
      "Processing sequence: 127400\n",
      "Processing sequence: 127500\n",
      "Processing sequence: 127600\n",
      "Processing sequence: 127700\n",
      "Processing sequence: 127800\n",
      "Processing sequence: 127900\n",
      "Processing sequence: 128000\n",
      "Processing sequence: 128100\n",
      "Processing sequence: 128200\n",
      "Processing sequence: 128300\n",
      "Processing sequence: 128400\n",
      "Processing sequence: 128500\n",
      "Processing sequence: 128600\n",
      "Processing sequence: 128700\n",
      "Processing sequence: 128800\n",
      "Processing sequence: 128900\n",
      "Processing sequence: 129000\n",
      "Processing sequence: 129100\n",
      "Processing sequence: 129200\n",
      "Processing sequence: 129300\n",
      "Processing sequence: 129400\n",
      "Processing sequence: 129500\n",
      "Processing sequence: 129600\n",
      "Processing sequence: 129700\n",
      "Processing sequence: 129800\n",
      "Processing sequence: 129900\n",
      "Processing sequence: 130000\n",
      "Processing sequence: 130100\n",
      "Processing sequence: 130200\n",
      "Processing sequence: 130300\n",
      "Processing sequence: 130400\n",
      "Processing sequence: 130500\n",
      "Processing sequence: 130600\n",
      "Processing sequence: 130700\n",
      "Processing sequence: 130800\n",
      "Processing sequence: 130900\n",
      "Processing sequence: 131000\n",
      "Processing sequence: 131100\n",
      "Processing sequence: 131200\n",
      "Processing sequence: 131300\n",
      "Processing sequence: 131400\n",
      "Processing sequence: 131500\n",
      "Processing sequence: 131600\n",
      "Processing sequence: 131700\n",
      "Processing sequence: 131800\n",
      "Processing sequence: 131900\n",
      "Processing sequence: 132000\n",
      "Processing sequence: 132100\n",
      "Processing sequence: 132200\n",
      "Processing sequence: 132300\n",
      "Processing sequence: 132400\n",
      "Processing sequence: 132500\n",
      "Processing sequence: 132600\n",
      "Processing sequence: 132700\n",
      "Processing sequence: 132800\n",
      "Processing sequence: 132900\n",
      "Processing sequence: 133000\n",
      "Processing sequence: 133100\n",
      "Processing sequence: 133200\n",
      "Processing sequence: 133300\n",
      "Processing sequence: 133400\n",
      "Processing sequence: 133500\n",
      "Processing sequence: 133600\n",
      "Processing sequence: 133700\n",
      "Processing sequence: 133800\n",
      "Processing sequence: 133900\n",
      "Processing sequence: 134000\n",
      "Processing sequence: 134100\n",
      "Processing sequence: 134200\n",
      "Processing sequence: 134300\n",
      "Processing sequence: 134400\n",
      "Processing sequence: 134500\n",
      "Processing sequence: 134600\n",
      "Processing sequence: 134700\n",
      "Processing sequence: 134800\n",
      "Processing sequence: 134900\n",
      "Processing sequence: 135000\n",
      "Processing sequence: 135100\n",
      "Processing sequence: 135200\n",
      "Processing sequence: 135300\n",
      "Processing sequence: 135400\n",
      "Processing sequence: 135500\n",
      "Processing sequence: 135600\n",
      "Processing sequence: 135700\n",
      "Processing sequence: 135800\n",
      "Processing sequence: 135900\n",
      "Processing sequence: 136000\n",
      "Processing sequence: 136100\n",
      "Processing sequence: 136200\n",
      "Processing sequence: 136300\n",
      "Processing sequence: 136400\n",
      "Processing sequence: 136500\n",
      "Processing sequence: 136600\n",
      "Processing sequence: 136700\n",
      "Processing sequence: 136800\n",
      "Processing sequence: 136900\n",
      "Processing sequence: 137000\n",
      "Processing sequence: 137100\n",
      "Processing sequence: 137200\n",
      "Processing sequence: 137300\n",
      "Processing sequence: 137400\n",
      "Processing sequence: 137500\n",
      "Processing sequence: 137600\n",
      "Processing sequence: 137700\n",
      "Processing sequence: 137800\n",
      "Processing sequence: 137900\n",
      "Processing sequence: 138000\n",
      "Processing sequence: 138100\n",
      "Processing sequence: 138200\n",
      "Processing sequence: 138300\n",
      "Processing sequence: 138400\n",
      "Processing sequence: 138500\n",
      "Processing sequence: 138600\n",
      "Processing sequence: 138700\n",
      "Processing sequence: 138800\n",
      "Processing sequence: 138900\n",
      "Processing sequence: 139000\n",
      "Processing sequence: 139100\n",
      "Processing sequence: 139200\n",
      "Processing sequence: 139300\n",
      "Processing sequence: 139400\n",
      "Processing sequence: 139500\n",
      "Processing sequence: 139600\n",
      "Processing sequence: 139700\n",
      "Processing sequence: 139800\n",
      "Processing sequence: 139900\n",
      "Processing sequence: 140000\n",
      "Processing sequence: 140100\n",
      "Processing sequence: 140200\n",
      "Processing sequence: 140300\n",
      "Processing sequence: 140400\n",
      "Processing sequence: 140500\n",
      "Processing sequence: 140600\n",
      "Processing sequence: 140700\n",
      "Processing sequence: 140800\n",
      "Processing sequence: 140900\n",
      "Processing sequence: 141000\n",
      "Processing sequence: 141100\n",
      "Processing sequence: 141200\n",
      "Processing sequence: 141300\n",
      "Processing sequence: 141400\n",
      "Processing sequence: 141500\n",
      "Processing sequence: 141600\n",
      "Processing sequence: 141700\n",
      "Processing sequence: 141800\n",
      "Processing sequence: 141900\n",
      "Processing sequence: 142000\n",
      "Processing sequence: 142100\n",
      "Processing sequence: 142200\n",
      "Processing sequence: 142300\n",
      "Processing sequence: 142400\n",
      "Processing sequence: 142500\n",
      "Processing sequence: 142600\n",
      "Processing sequence: 142700\n",
      "Processing sequence: 142800\n",
      "Processing sequence: 142900\n",
      "Processing sequence: 143000\n",
      "Processing sequence: 143100\n",
      "Processing sequence: 143200\n",
      "Processing sequence: 143300\n",
      "Processing sequence: 143400\n",
      "Processing sequence: 143500\n",
      "Processing sequence: 143600\n",
      "Processing sequence: 143700\n",
      "Processing sequence: 143800\n",
      "Processing sequence: 143900\n",
      "Processing sequence: 144000\n",
      "Processing sequence: 144100\n",
      "Processing sequence: 144200\n",
      "Processing sequence: 144300\n",
      "Processing sequence: 144400\n",
      "Processing sequence: 144500\n",
      "Processing sequence: 144600\n",
      "Processing sequence: 144700\n",
      "Processing sequence: 144800\n",
      "Processing sequence: 144900\n",
      "Processing sequence: 145000\n",
      "Processing sequence: 145100\n",
      "Processing sequence: 145200\n",
      "Processing sequence: 145300\n",
      "Processing sequence: 145400\n",
      "Processing sequence: 145500\n",
      "Processing sequence: 145600\n",
      "Processing sequence: 145700\n",
      "Processing sequence: 145800\n",
      "Processing sequence: 145900\n",
      "Processing sequence: 146000\n",
      "Processing sequence: 146100\n",
      "Processing sequence: 146200\n",
      "Processing sequence: 146300\n",
      "Processing sequence: 146400\n",
      "Processing sequence: 146500\n",
      "Processing sequence: 146600\n",
      "Processing sequence: 146700\n",
      "Processing sequence: 146800\n",
      "Processing sequence: 146900\n",
      "Processing sequence: 147000\n",
      "Processing sequence: 147100\n",
      "Processing sequence: 147200\n",
      "Processing sequence: 147300\n",
      "Processing sequence: 147400\n",
      "Processing sequence: 147500\n",
      "Processing sequence: 147600\n",
      "Processing sequence: 147700\n",
      "Processing sequence: 147800\n",
      "Processing sequence: 147900\n",
      "Processing sequence: 148000\n",
      "Processing sequence: 148100\n",
      "Processing sequence: 148200\n",
      "Processing sequence: 148300\n",
      "Processing sequence: 148400\n",
      "Processing sequence: 148500\n",
      "Processing sequence: 148600\n",
      "Processing sequence: 148700\n",
      "Processing sequence: 148800\n",
      "Processing sequence: 148900\n",
      "Processing sequence: 149000\n",
      "Processing sequence: 149100\n",
      "Processing sequence: 149200\n",
      "Processing sequence: 149300\n",
      "Processing sequence: 149400\n",
      "Processing sequence: 149500\n",
      "Processing sequence: 149600\n",
      "Processing sequence: 149700\n",
      "Processing sequence: 149800\n",
      "Processing sequence: 149900\n",
      "Processing sequence: 150000\n",
      "Processing sequence: 150100\n",
      "Processing sequence: 150200\n",
      "Processing sequence: 150300\n",
      "Processing sequence: 150400\n",
      "Processing sequence: 150500\n",
      "Processing sequence: 150600\n",
      "Processing sequence: 150700\n",
      "Processing sequence: 150800\n",
      "Processing sequence: 150900\n",
      "Processing sequence: 151000\n",
      "Processing sequence: 151100\n",
      "Processing sequence: 151200\n",
      "Processing sequence: 151300\n",
      "Processing sequence: 151400\n",
      "Processing sequence: 151500\n",
      "Processing sequence: 151600\n",
      "Processing sequence: 151700\n",
      "Processing sequence: 151800\n",
      "Processing sequence: 151900\n",
      "Processing sequence: 152000\n",
      "Processing sequence: 152100\n",
      "Processing sequence: 152200\n",
      "Processing sequence: 152300\n",
      "Processing sequence: 152360\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.fasta\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.bad.accnos\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.contigs.good.groups\n",
      "\n",
      "\n",
      "It took 2 secs to screen 152360 sequences.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.screen.seqs(fasta='current', group='current', maxambig=0, maxlength=275)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > unique.seqs(fasta=stability.trim.contigs.good.fasta)\n",
      "1000\t351\n",
      "2000\t590\n",
      "3000\t784\n",
      "4000\t991\n",
      "5000\t1196\n",
      "6000\t1400\n",
      "7000\t1611\n",
      "8000\t1786\n",
      "9000\t1952\n",
      "10000\t2101\n",
      "11000\t2270\n",
      "12000\t2433\n",
      "13000\t2605\n",
      "14000\t2767\n",
      "15000\t2934\n",
      "16000\t3086\n",
      "17000\t3260\n",
      "18000\t3416\n",
      "19000\t3563\n",
      "20000\t3705\n",
      "21000\t3862\n",
      "22000\t3995\n",
      "23000\t4139\n",
      "24000\t4263\n",
      "25000\t4400\n",
      "26000\t4525\n",
      "27000\t4660\n",
      "28000\t4807\n",
      "29000\t4952\n",
      "30000\t5116\n",
      "31000\t5272\n",
      "32000\t5419\n",
      "33000\t5554\n",
      "34000\t5709\n",
      "35000\t5856\n",
      "36000\t5987\n",
      "37000\t6110\n",
      "38000\t6241\n",
      "39000\t6375\n",
      "40000\t6521\n",
      "41000\t6645\n",
      "42000\t6798\n",
      "43000\t6915\n",
      "44000\t7019\n",
      "45000\t7153\n",
      "46000\t7280\n",
      "47000\t7416\n",
      "48000\t7529\n",
      "49000\t7642\n",
      "50000\t7762\n",
      "51000\t7889\n",
      "52000\t7998\n",
      "53000\t8109\n",
      "54000\t8240\n",
      "55000\t8359\n",
      "56000\t8476\n",
      "57000\t8603\n",
      "58000\t8738\n",
      "59000\t8863\n",
      "60000\t8992\n",
      "61000\t9112\n",
      "62000\t9234\n",
      "63000\t9356\n",
      "64000\t9492\n",
      "65000\t9616\n",
      "66000\t9708\n",
      "67000\t9816\n",
      "68000\t9943\n",
      "69000\t10079\n",
      "70000\t10235\n",
      "71000\t10366\n",
      "72000\t10509\n",
      "73000\t10646\n",
      "74000\t10798\n",
      "75000\t10948\n",
      "76000\t11087\n",
      "77000\t11219\n",
      "78000\t11325\n",
      "79000\t11442\n",
      "80000\t11541\n",
      "81000\t11664\n",
      "82000\t11758\n",
      "83000\t11852\n",
      "84000\t11961\n",
      "85000\t12045\n",
      "86000\t12147\n",
      "87000\t12236\n",
      "88000\t12340\n",
      "89000\t12442\n",
      "90000\t12533\n",
      "91000\t12650\n",
      "92000\t12756\n",
      "93000\t12854\n",
      "94000\t12953\n",
      "95000\t13040\n",
      "96000\t13133\n",
      "97000\t13214\n",
      "98000\t13287\n",
      "99000\t13372\n",
      "100000\t13485\n",
      "101000\t13586\n",
      "102000\t13717\n",
      "103000\t13817\n",
      "104000\t13913\n",
      "105000\t14021\n",
      "106000\t14123\n",
      "107000\t14230\n",
      "108000\t14324\n",
      "109000\t14423\n",
      "110000\t14496\n",
      "111000\t14581\n",
      "112000\t14663\n",
      "113000\t14723\n",
      "114000\t14794\n",
      "115000\t14892\n",
      "116000\t14994\n",
      "117000\t15097\n",
      "118000\t15204\n",
      "119000\t15311\n",
      "120000\t15409\n",
      "121000\t15515\n",
      "122000\t15632\n",
      "123000\t15756\n",
      "124000\t15861\n",
      "125000\t15969\n",
      "126000\t16100\n",
      "127000\t16217\n",
      "128000\t16326\n",
      "128872\t16426\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.names\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.fasta\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.unique.seqs(fasta='stability.trim.contigs.good.fasta')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > count.seqs(name=stability.trim.contigs.good.names,group=stability.contigs.good.groups)\n",
      "\n",
      "Using 1 processors.\n",
      "It took 0 secs to create a table for 128872 sequences.\n",
      "\n",
      "\n",
      "Total number of sequences: 128872\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.count_table\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.count.seqs(name='stability.trim.contigs.good.names', group='stability.contigs.good.groups')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > summary.seqs(count=stability.trim.contigs.good.count_table)\n",
      "Using C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.fasta as input file for the fasta parameter.\n",
      "\n",
      "Using 1 processors.\n",
      "\n",
      "\t\tStart\tEnd\tNBases\tAmbigs\tPolymer\tNumSeqs\n",
      "Minimum:\t1\t250\t250\t0\t3\t1\n",
      "2.5%-tile:\t1\t252\t252\t0\t3\t3222\n",
      "25%-tile:\t1\t252\t252\t0\t4\t32219\n",
      "Median: \t1\t252\t252\t0\t4\t64437\n",
      "75%-tile:\t1\t253\t253\t0\t5\t96655\n",
      "97.5%-tile:\t1\t253\t253\t0\t6\t125651\n",
      "Maximum:\t1\t270\t270\t0\t12\t128872\n",
      "Mean:\t1\t252.462\t252.462\t0\t4.36693\n",
      "# of unique seqs:\t16426\n",
      "total # of seqs:\t128872\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.summary\n",
      "\n",
      "It took 1 secs to summarize 128872 sequences.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.summary.seqs(count='stability.trim.contigs.good.count_table')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > pcr.seqs(fasta=silva.bacteria\\silva.bacteria.fasta,start=11894,end=25319,keepdots=F,processors=4)\n",
      "\n",
      "Using 4 processors.\n",
      "Processing sequence: 100\n",
      "Processing sequence: 100\n",
      "Processing sequence: 100\n",
      "Processing sequence: 100\n",
      "Processing sequence: 200\n",
      "Processing sequence: 200\n",
      "Processing sequence: 200\n",
      "Processing sequence: 200\n",
      "Processing sequence: 300\n",
      "Processing sequence: 300\n",
      "Processing sequence: 300\n",
      "Processing sequence: 300\n",
      "Processing sequence: 400\n",
      "Processing sequence: 400\n",
      "Processing sequence: 400\n",
      "Processing sequence: 400\n",
      "Processing sequence: 500\n",
      "Processing sequence: 500\n",
      "Processing sequence: 500\n",
      "Processing sequence: 500\n",
      "Processing sequence: 600\n",
      "Processing sequence: 600\n",
      "Processing sequence: 600\n",
      "Processing sequence: 600\n",
      "Processing sequence: 700\n",
      "Processing sequence: 700\n",
      "Processing sequence: 700\n",
      "Processing sequence: 700\n",
      "Processing sequence: 800\n",
      "Processing sequence: 800\n",
      "Processing sequence: 800\n",
      "Processing sequence: 800\n",
      "Processing sequence: 900\n",
      "Processing sequence: 900\n",
      "Processing sequence: 900\n",
      "Processing sequence: 900\n",
      "Processing sequence: 1000\n",
      "Processing sequence: 1000\n",
      "Processing sequence: 1000\n",
      "Processing sequence: 1000\n",
      "Processing sequence: 1100\n",
      "Processing sequence: 1100\n",
      "Processing sequence: 1100\n",
      "Processing sequence: 1100\n",
      "Processing sequence: 1200\n",
      "Processing sequence: 1200\n",
      "Processing sequence: 1200\n",
      "Processing sequence: 1200\n",
      "Processing sequence: 1300\n",
      "Processing sequence: 1300\n",
      "Processing sequence: 1300\n",
      "Processing sequence: 1300\n",
      "Processing sequence: 1400\n",
      "Processing sequence: 1400\n",
      "Processing sequence: 1400\n",
      "Processing sequence: 1400\n",
      "Processing sequence: 1500\n",
      "Processing sequence: 1500\n",
      "Processing sequence: 1500\n",
      "Processing sequence: 1500\n",
      "Processing sequence: 1600\n",
      "Processing sequence: 1600\n",
      "Processing sequence: 1600\n",
      "Processing sequence: 1600\n",
      "Processing sequence: 1700\n",
      "Processing sequence: 1700\n",
      "Processing sequence: 1700\n",
      "Processing sequence: 1700\n",
      "Processing sequence: 1800\n",
      "Processing sequence: 1800\n",
      "Processing sequence: 1800\n",
      "Processing sequence: 1800\n",
      "Processing sequence: 1900\n",
      "Processing sequence: 1900\n",
      "Processing sequence: 1900\n",
      "Processing sequence: 1900\n",
      "Processing sequence: 2000\n",
      "Processing sequence: 2000\n",
      "Processing sequence: 2000\n",
      "Processing sequence: 2000\n",
      "Processing sequence: 2100\n",
      "Processing sequence: 2100\n",
      "Processing sequence: 2100\n",
      "Processing sequence: 2100\n",
      "Processing sequence: 2200\n",
      "Processing sequence: 2200\n",
      "Processing sequence: 2200\n",
      "Processing sequence: 2200\n",
      "Processing sequence: 2300\n",
      "Processing sequence: 2300\n",
      "Processing sequence: 2300\n",
      "Processing sequence: 2300\n",
      "Processing sequence: 2400\n",
      "Processing sequence: 2400\n",
      "Processing sequence: 2400\n",
      "Processing sequence: 2400\n",
      "Processing sequence: 2500\n",
      "Processing sequence: 2500\n",
      "Processing sequence: 2500\n",
      "Processing sequence: 2500\n",
      "Processing sequence: 2600\n",
      "Processing sequence: 2600\n",
      "Processing sequence: 2600\n",
      "Processing sequence: 2600\n",
      "Processing sequence: 2700\n",
      "Processing sequence: 2700\n",
      "Processing sequence: 2700\n",
      "Processing sequence: 2700\n",
      "Processing sequence: 2800\n",
      "Processing sequence: 2800\n",
      "Processing sequence: 2800\n",
      "Processing sequence: 2800\n",
      "Processing sequence: 2900\n",
      "Processing sequence: 2900\n",
      "Processing sequence: 2900\n",
      "Processing sequence: 2900\n",
      "Processing sequence: 3000\n",
      "Processing sequence: 3000\n",
      "Processing sequence: 3000\n",
      "Processing sequence: 3000\n",
      "Processing sequence: 3100\n",
      "Processing sequence: 3100\n",
      "Processing sequence: 3100\n",
      "Processing sequence: 3100\n",
      "Processing sequence: 3200\n",
      "Processing sequence: 3200\n",
      "Processing sequence: 3200\n",
      "Processing sequence: 3200\n",
      "Processing sequence: 3300\n",
      "Processing sequence: 3300\n",
      "Processing sequence: 3300\n",
      "Processing sequence: 3300\n",
      "Processing sequence: 3400\n",
      "Processing sequence: 3400\n",
      "Processing sequence: 3400\n",
      "Processing sequence: 3400\n",
      "Processing sequence: 3500\n",
      "Processing sequence: 3500\n",
      "Processing sequence: 3500\n",
      "Processing sequence: 3500\n",
      "Processing sequence: 3600\n",
      "Processing sequence: 3600\n",
      "Processing sequence: 3600\n",
      "Processing sequence: 3600\n",
      "Processing sequence: 3700\n",
      "Processing sequence: 3700\n",
      "Processing sequence: 3700\n",
      "Thread Processing sequence: 3739\n",
      "Processing sequence: 3700\n",
      "Thread Processing sequence: 3739\n",
      "Thread Processing sequence: 3739\n",
      "Processing sequence: 3739\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\silva.bacteria.pcr.fasta\n",
      "\n",
      "\n",
      "It took 10 secs to screen 14956 sequences.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.pcr.seqs(fasta='silva.bacteria\\\\silva.bacteria.fasta', start=11894, end=25319, keepdots='F', processors=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > rename.file(input=silva.bacteria.pcr.fasta,new=silva.v4.fasta)\n",
      "\n",
      "Current files saved by mothur:\n",
      "fasta=C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\silva.bacteria.pcr.fasta\n",
      "group=C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.contigs.good.groups\n",
      "name=C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.names\n",
      "qfile=C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.qual\n",
      "contigsreport=C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.contigs.report\n",
      "count=C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.count_table\n",
      "processors=4\n",
      "summary=C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.summary\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.rename.file(input='silva.bacteria.pcr.fasta', new='silva.v4.fasta')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > summary.seqs(fasta=silva.v4.fasta)\n",
      "\n",
      "Using 4 processors.\n",
      "\n",
      "\t\tStart\tEnd\tNBases\tAmbigs\tPolymer\tNumSeqs\n",
      "Minimum:\t1\t13424\t270\t0\t3\t1\n",
      "2.5%-tile:\t1\t13425\t292\t0\t4\t374\n",
      "25%-tile:\t1\t13425\t293\t0\t4\t3740\n",
      "Median: \t1\t13425\t293\t0\t4\t7479\n",
      "75%-tile:\t1\t13425\t293\t0\t5\t11218\n",
      "97.5%-tile:\t1\t13425\t294\t1\t6\t14583\n",
      "Maximum:\t3\t13425\t351\t5\t9\t14956\n",
      "Mean:\t1.00074\t13425\t292.977\t0.0573014\t4.57014\n",
      "# of Seqs:\t14956\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\silva.v4.summary\n",
      "\n",
      "It took 3 secs to summarize 14956 sequences.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.summary.seqs(fasta='silva.v4.fasta')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > align.seqs(fasta=stability.trim.contigs.good.unique.fasta,reference=silva.v4.fasta)\n",
      "\n",
      "Using 4 processors.\n",
      "\n",
      "Reading in the C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\silva.v4.fasta template sequences...\tDONE.\n",
      "It took 4 to read  14956 sequences.\n",
      "Aligning sequences from C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.fasta ...\n",
      "\n",
      "Reading in the C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\silva.v4.fasta template sequences...\t\n",
      "Reading in the C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\silva.v4.fasta template sequences...\t\n",
      "Reading in the C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\silva.v4.fasta template sequences...\t100\n",
      "200\n",
      "300\n",
      "400\n",
      "500\n",
      "600\n",
      "700\n",
      "800\n",
      "900\n",
      "1000\n",
      "1100\n",
      "1200\n",
      "1300\n",
      "1400\n",
      "1500\n",
      "1600\n",
      "1700\n",
      "1800\n",
      "1900\n",
      "2000\n",
      "2100\n",
      "2200\n",
      "2300\n",
      "2400\n",
      "2500\n",
      "2600\n",
      "2700\n",
      "2800\n",
      "2900\n",
      "3000\n",
      "3100\n",
      "3200\n",
      "3300\n",
      "DONE.\n",
      "It took 4 to read  14956 sequences.\n",
      "DONE.\n",
      "It took 4 to read  14956 sequences.\n",
      "DONE.\n",
      "It took 4 to read  14956 sequences.\n",
      "3400\n",
      "3500\n",
      "3600\n",
      "3700\n",
      "3800\n",
      "3900\n",
      "100\n",
      "100\n",
      "100\n",
      "4000\n",
      "200\n",
      "200\n",
      "4100\n",
      "200\n",
      "4108\n",
      "300\n",
      "300\n",
      "300\n",
      "400\n",
      "400\n",
      "400\n",
      "500\n",
      "500\n",
      "500\n",
      "600\n",
      "600\n",
      "600\n",
      "700\n",
      "700\n",
      "700\n",
      "800\n",
      "800\n",
      "800\n",
      "900\n",
      "900\n",
      "900\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "1100\n",
      "1100\n",
      "1100\n",
      "1200\n",
      "1200\n",
      "1200\n",
      "1300\n",
      "1300\n",
      "1300\n",
      "1400\n",
      "1400\n",
      "1400\n",
      "1500\n",
      "1500\n",
      "1500\n",
      "1600\n",
      "1600\n",
      "1600\n",
      "1700\n",
      "1700\n",
      "1700\n",
      "1800\n",
      "1800\n",
      "1800\n",
      "1900\n",
      "1900\n",
      "1900\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "2100\n",
      "2100\n",
      "2100\n",
      "2200\n",
      "2200\n",
      "2200\n",
      "2300\n",
      "2300\n",
      "2300\n",
      "2400\n",
      "2400\n",
      "2400\n",
      "2500\n",
      "2500\n",
      "2500\n",
      "2600\n",
      "2600\n",
      "2600\n",
      "2700\n",
      "2700\n",
      "2700\n",
      "2800\n",
      "2800\n",
      "2800\n",
      "2900\n",
      "2900\n",
      "2900\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3100\n",
      "3100\n",
      "3100\n",
      "3200\n",
      "3200\n",
      "3200\n",
      "3300\n",
      "3300\n",
      "3300\n",
      "3400\n",
      "3400\n",
      "3400\n",
      "3500\n",
      "3500\n",
      "3500\n",
      "3600\n",
      "3600\n",
      "3700\n",
      "3600\n",
      "3700\n",
      "3800\n",
      "3700\n",
      "3800\n",
      "3900\n",
      "3800\n",
      "3900\n",
      "4000\n",
      "3900\n",
      "4000\n",
      "4100\n",
      "4106\n",
      "4000\n",
      "4100\n",
      "4106\n",
      "4100\n",
      "4106\n",
      "It took 11 secs to align 16426 sequences.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.align\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.align.report\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.align.seqs(fasta='stability.trim.contigs.good.unique.fasta', reference='silva.v4.fasta')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > summary.seqs(fasta=stability.trim.contigs.good.unique.align,count=stability.trim.contigs.good.count_table)\n",
      "\n",
      "Using 4 processors.\n",
      "\n",
      "\t\tStart\tEnd\tNBases\tAmbigs\tPolymer\tNumSeqs\n",
      "Minimum:\t1250\t10693\t250\t0\t3\t1\n",
      "2.5%-tile:\t1968\t11550\t252\t0\t3\t3222\n",
      "25%-tile:\t1968\t11550\t252\t0\t4\t32219\n",
      "Median: \t1968\t11550\t252\t0\t4\t64437\n",
      "75%-tile:\t1968\t11550\t253\t0\t5\t96655\n",
      "97.5%-tile:\t1968\t11550\t253\t0\t6\t125651\n",
      "Maximum:\t1982\t13400\t270\t0\t12\t128872\n",
      "Mean:\t1967.99\t11550\t252.462\t0\t4.36693\n",
      "# of unique seqs:\t16426\n",
      "total # of seqs:\t128872\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.summary\n",
      "\n",
      "It took 3 secs to summarize 128872 sequences.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.summary.seqs(fasta='stability.trim.contigs.good.unique.align', count='stability.trim.contigs.good.count_table')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > filter.seqs(fasta=stability.trim.contigs.good.unique.align,vertical=T,trump=.)\n",
      "\n",
      "Using 4 processors.\n",
      "Creating Filter... \n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "100\n",
      "100\n",
      "100\n",
      "100\n",
      "200\n",
      "200\n",
      "200\n",
      "200\n",
      "300\n",
      "300\n",
      "300\n",
      "300\n",
      "400\n",
      "400\n",
      "400\n",
      "400\n",
      "500\n",
      "500\n",
      "500\n",
      "500\n",
      "600\n",
      "600\n",
      "600\n",
      "600\n",
      "700\n",
      "700\n",
      "700\n",
      "700\n",
      "800\n",
      "800\n",
      "800\n",
      "800\n",
      "900\n",
      "900\n",
      "900\n",
      "900\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "1100\n",
      "1100\n",
      "1100\n",
      "1100\n",
      "1200\n",
      "1200\n",
      "1200\n",
      "1200\n",
      "1300\n",
      "1300\n",
      "1300\n",
      "1300\n",
      "1400\n",
      "1400\n",
      "1400\n",
      "1400\n",
      "1500\n",
      "1500\n",
      "1500\n",
      "1500\n",
      "1600\n",
      "1600\n",
      "1600\n",
      "1600\n",
      "1700\n",
      "1700\n",
      "1700\n",
      "1700\n",
      "1800\n",
      "1800\n",
      "1800\n",
      "1800\n",
      "1900\n",
      "1900\n",
      "1900\n",
      "1900\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "2100\n",
      "2100\n",
      "2100\n",
      "2100\n",
      "2200\n",
      "2200\n",
      "2200\n",
      "2200\n",
      "2300\n",
      "2300\n",
      "2300\n",
      "2300\n",
      "2400\n",
      "2400\n",
      "2400\n",
      "2400\n",
      "2500\n",
      "2500\n",
      "2500\n",
      "2500\n",
      "2600\n",
      "2600\n",
      "2600\n",
      "2600\n",
      "2700\n",
      "2700\n",
      "2700\n",
      "2800\n",
      "2700\n",
      "2800\n",
      "2800\n",
      "2800\n",
      "2900\n",
      "2900\n",
      "2900\n",
      "2900\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3100\n",
      "3100\n",
      "3100\n",
      "3100\n",
      "3200\n",
      "3200\n",
      "3200\n",
      "3200\n",
      "3300\n",
      "3300\n",
      "3300\n",
      "3300\n",
      "3400\n",
      "3400\n",
      "3400\n",
      "3400\n",
      "3500\n",
      "3500\n",
      "3500\n",
      "3500\n",
      "3600\n",
      "3600\n",
      "3600\n",
      "3600\n",
      "3700\n",
      "3700\n",
      "3700\n",
      "3700\n",
      "3800\n",
      "3800\n",
      "3800\n",
      "3800\n",
      "3900\n",
      "3900\n",
      "3900\n",
      "3900\n",
      "4000\n",
      "4000\n",
      "4000\n",
      "4000\n",
      "4100\n",
      "4100\n",
      "4106\n",
      "4106\n",
      "4100\n",
      "4108\n",
      "4100\n",
      "4106\n",
      "\n",
      "\n",
      "Running Filter... \n",
      "0\n",
      "0\n",
      "0\n",
      "100\n",
      "100\n",
      "100\n",
      "100\n",
      "200\n",
      "200\n",
      "200\n",
      "200\n",
      "300\n",
      "300\n",
      "300\n",
      "300\n",
      "400\n",
      "400\n",
      "400\n",
      "400\n",
      "500\n",
      "500\n",
      "500\n",
      "500\n",
      "600\n",
      "600\n",
      "600\n",
      "600\n",
      "700\n",
      "700\n",
      "700\n",
      "700\n",
      "800\n",
      "800\n",
      "800\n",
      "800\n",
      "900\n",
      "900\n",
      "900\n",
      "900\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "1100\n",
      "1100\n",
      "1100\n",
      "1100\n",
      "1200\n",
      "1200\n",
      "1200\n",
      "1200\n",
      "1300\n",
      "1300\n",
      "1300\n",
      "1300\n",
      "1400\n",
      "1400\n",
      "1400\n",
      "1400\n",
      "1500\n",
      "1500\n",
      "1500\n",
      "1500\n",
      "1600\n",
      "1600\n",
      "1600\n",
      "1600\n",
      "1700\n",
      "1700\n",
      "1700\n",
      "1700\n",
      "1800\n",
      "1800\n",
      "1800\n",
      "1800\n",
      "1900\n",
      "1900\n",
      "1900\n",
      "1900\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "2100\n",
      "2100\n",
      "2100\n",
      "2100\n",
      "2200\n",
      "2200\n",
      "2200\n",
      "2200\n",
      "2300\n",
      "2300\n",
      "2300\n",
      "2300\n",
      "2400\n",
      "2400\n",
      "2400\n",
      "2400\n",
      "2500\n",
      "2500\n",
      "2500\n",
      "2500\n",
      "2600\n",
      "2600\n",
      "2600\n",
      "2600\n",
      "2700\n",
      "2700\n",
      "2700\n",
      "2700\n",
      "2800\n",
      "2800\n",
      "2800\n",
      "2800\n",
      "2900\n",
      "2900\n",
      "2900\n",
      "2900\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3100\n",
      "3000\n",
      "3100\n",
      "3100\n",
      "3100\n",
      "3200\n",
      "3200\n",
      "3200\n",
      "3200\n",
      "3300\n",
      "3300\n",
      "3300\n",
      "3300\n",
      "3400\n",
      "3400\n",
      "3400\n",
      "3500\n",
      "3400\n",
      "3500\n",
      "3500\n",
      "3600\n",
      "3500\n",
      "3600\n",
      "3600\n",
      "3600\n",
      "3700\n",
      "3700\n",
      "3700\n",
      "3700\n",
      "3800\n",
      "3800\n",
      "3800\n",
      "3800\n",
      "3900\n",
      "3900\n",
      "3900\n",
      "3900\n",
      "4000\n",
      "4000\n",
      "4000\n",
      "4100\n",
      "4000\n",
      "4108\n",
      "4100\n",
      "4100\n",
      "4106\n",
      "4106\n",
      "4100\n",
      "4106\n",
      "\n",
      "\n",
      "\n",
      "Length of filtered alignment: 369\n",
      "Number of columns removed: 13056\n",
      "Length of the original alignment: 13425\n",
      "Number of sequences used to construct filter: 16426\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.filter\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.fasta\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.filter.seqs(fasta='stability.trim.contigs.good.unique.align', vertical='T', trump='.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > unique.seqs(fasta=stability.trim.contigs.good.unique.filter.fasta,count=stability.trim.contigs.good.count_table)\n",
      "1000\t960\n",
      "2000\t1913\n",
      "3000\t2878\n",
      "4000\t3844\n",
      "5000\t4804\n",
      "6000\t5743\n",
      "7000\t6691\n",
      "8000\t7647\n",
      "9000\t8610\n",
      "10000\t9571\n",
      "11000\t10514\n",
      "12000\t11460\n",
      "13000\t12400\n",
      "14000\t13357\n",
      "15000\t14309\n",
      "16000\t15258\n",
      "16426\t15647\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.count_table\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.fasta\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.unique.seqs(fasta='stability.trim.contigs.good.unique.filter.fasta', count='stability.trim.contigs.good.count_table')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > pre.cluster(fasta=stability.trim.contigs.good.unique.filter.unique.fasta,count=stability.trim.contigs.good.unique.filter.count_table,diffs=2)\n",
      "\n",
      "Using 4 processors.\n",
      "\n",
      "Processing group F3D0:\n",
      "\n",
      "Processing group F3D149:\n",
      "\n",
      "Processing group F3D6:\n",
      "\n",
      "Processing group F3D144:\n",
      "0\t1417\t47\n",
      "0\t898\t37\n",
      "100\t500\t435\n",
      "0\t1272\t81\n",
      "200\t450\t485\n",
      "300\t439\t496\n",
      "400\t434\t501\n",
      "500\t431\t504\n",
      "600\t426\t509\n",
      "700\t419\t516\n",
      "800\t419\t516\n",
      "900\t418\t517\n",
      "100\t626\t727\n",
      "200\t567\t786\n",
      "300\t543\t810\n",
      "400\t538\t815\n",
      "500\t532\t821\n",
      "935\t418\t517\n",
      "Total number of sequences before pre.cluster was 935.\n",
      "pre.cluster removed 517 sequences.\n",
      "\n",
      "600\t519\t834\n",
      "700\t513\t840\n",
      "100\t710\t754\n",
      "800\t509\t844\n",
      "900\t501\t852\n",
      "1000\t495\t858\n",
      "1100\t493\t860\n",
      "1200\t493\t860\n",
      "1300\t493\t860\n",
      "1353\t493\t860\n",
      "Total number of sequences before pre.cluster was 1353.\n",
      "pre.cluster removed 860 sequences.\n",
      "\n",
      "0\t2123\t84\n",
      "200\t641\t823\n",
      "It took 0 secs to cluster 935 sequences.\n",
      "\n",
      "Processing group F3D145:\n",
      "300\t607\t857\n",
      "400\t598\t866\n",
      "100\t1091\t1116\n",
      "500\t587\t877\n",
      "600\t584\t880\n",
      "700\t580\t884\n",
      "200\t963\t1244\n",
      "800\t572\t892\n",
      "It took 0 secs to cluster 1353 sequences.\n",
      "\n",
      "Processing group F3D7:\n",
      "900\t568\t896\n",
      "300\t913\t1294\n",
      "1000\t563\t901\n",
      "400\t885\t1322\n",
      "1100\t562\t902\n",
      "1200\t558\t906\n",
      "1300\t558\t906\n",
      "500\t871\t1336\n",
      "1400\t558\t906\n",
      "600\t862\t1345\n",
      "700\t860\t1347\n",
      "1464\t558\t906\n",
      "Total number of sequences before pre.cluster was 1464.\n",
      "pre.cluster removed 906 sequences.\n",
      "\n",
      "800\t856\t1351\n",
      "900\t854\t1353\n",
      "1000\t845\t1362\n",
      "1100\t842\t1365\n",
      "1200\t838\t1369\n",
      "1300\t836\t1371\n",
      "1400\t832\t1375\n",
      "1500\t824\t1383\n",
      "1600\t822\t1385\n",
      "1700\t819\t1388\n",
      "1800\t819\t1388\n",
      "1900\t818\t1389\n",
      "2000\t817\t1390\n",
      "2100\t817\t1390\n",
      "2200\t817\t1390\n",
      "2207\t817\t1390\n",
      "Total number of sequences before pre.cluster was 2207.\n",
      "pre.cluster removed 1390 sequences.\n",
      "\n",
      "0\t1254\t62\n",
      "0\t790\t51\n",
      "100\t395\t446\n",
      "200\t365\t476\n",
      "300\t354\t487\n",
      "400\t346\t495\n",
      "100\t641\t675\n",
      "500\t337\t504\n",
      "600\t335\t506\n",
      "700\t330\t511\n",
      "800\t329\t512\n",
      "200\t584\t732\n",
      "300\t558\t758\n",
      "400\t550\t766\n",
      "841\t329\t512\n",
      "Total number of sequences before pre.cluster was 841.\n",
      "pre.cluster removed 512 sequences.\n",
      "\n",
      "500\t541\t775\n",
      "600\t534\t782\n",
      "700\t528\t788\n",
      "800\t523\t793\n",
      "900\t518\t798\n",
      "1000\t514\t802\n",
      "1100\t510\t806\n",
      "1200\t509\t807\n",
      "1300\t509\t807\n",
      "It took 0 secs to cluster 2207 sequences.\n",
      "\n",
      "Processing group F3D150:\n",
      "1316\t509\t807\n",
      "Total number of sequences before pre.cluster was 1316.\n",
      "pre.cluster removed 807 sequences.\n",
      "\n",
      "It took 0 secs to cluster 1464 sequences.\n",
      "\n",
      "Processing group F3D1:\n",
      "It took 0 secs to cluster 841 sequences.\n",
      "\n",
      "Processing group F3D8:\n",
      "It took 0 secs to cluster 1316 sequences.\n",
      "\n",
      "Processing group F3D146:\n",
      "0\t1179\t48\n",
      "0\t1103\t42\n",
      "0\t934\t43\n",
      "100\t674\t553\n",
      "100\t499\t478\n",
      "200\t595\t632\n",
      "200\t458\t519\n",
      "300\t441\t536\n",
      "300\t569\t658\n",
      "400\t434\t543\n",
      "400\t559\t668\n",
      "0\t1074\t29\n",
      "500\t433\t544\n",
      "600\t429\t548\n",
      "500\t559\t668\n",
      "700\t424\t553\n",
      "800\t422\t555\n",
      "900\t420\t557\n",
      "600\t552\t675\n",
      "700\t550\t677\n",
      "800\t544\t683\n",
      "900\t538\t689\n",
      "100\t581\t564\n",
      "1000\t533\t694\n",
      "1100\t533\t694\n",
      "1200\t532\t695\n",
      "977\t420\t557\n",
      "Total number of sequences before pre.cluster was 977.\n",
      "pre.cluster removed 557 sequences.\n",
      "\n",
      "100\t652\t451\n",
      "200\t574\t529\n",
      "300\t557\t546\n",
      "1227\t532\t695\n",
      "Total number of sequences before pre.cluster was 1227.\n",
      "pre.cluster removed 695 sequences.\n",
      "\n",
      "400\t546\t557\n",
      "200\t526\t619\n",
      "500\t543\t560\n",
      "600\t540\t563\n",
      "700\t534\t569\n",
      "800\t528\t575\n",
      "300\t499\t646\n",
      "900\t524\t579\n",
      "1000\t523\t580\n",
      "1100\t523\t580\n",
      "400\t492\t653\n",
      "500\t489\t656\n",
      "600\t486\t659\n",
      "It took 0 secs to cluster 977 sequences.\n",
      "\n",
      "Processing group F3D9:\n",
      "It took 0 secs to cluster 1227 sequences.\n",
      "700\t481\t664\n",
      "\n",
      "Processing group F3D2:\n",
      "800\t480\t665\n",
      "900\t477\t668\n",
      "1000\t476\t669\n",
      "1100\t475\t670\n",
      "1145\t475\t670\n",
      "Total number of sequences before pre.cluster was 1145.\n",
      "pre.cluster removed 670 sequences.\n",
      "\n",
      "1103\t523\t580\n",
      "Total number of sequences before pre.cluster was 1103.\n",
      "pre.cluster removed 580 sequences.\n",
      "\n",
      "0\t1205\t56\n",
      "100\t649\t612\n",
      "200\t572\t689\n",
      "300\t557\t704\n",
      "400\t549\t712\n",
      "500\t537\t724\n",
      "600\t534\t727\n",
      "It took 0 secs to cluster 1103 sequences.\n",
      "\n",
      "Processing group F3D147:\n",
      "700\t529\t732\n",
      "800\t525\t736\n",
      "900\t523\t738\n",
      "1000\t521\t740\n",
      "1100\t521\t740\n",
      "1200\t520\t741\n",
      "1261\t519\t742\n",
      "Total number of sequences before pre.cluster was 1261.\n",
      "pre.cluster removed 742 sequences.\n",
      "\n",
      "It took 0 secs to cluster 1145 sequences.\n",
      "\n",
      "Processing group F3D141:\n",
      "0\t2425\t210\n",
      "0\t1040\t47\n",
      "It took 0 secs to cluster 1261 sequences.\n",
      "\n",
      "Processing group Mock:\n",
      "100\t1132\t1503\n",
      "0\t442\t52\n",
      "200\t1029\t1606\n",
      "100\t128\t366\n",
      "200\t125\t369\n",
      "300\t121\t373\n",
      "400\t118\t376\n",
      "300\t965\t1670\n",
      "400\t929\t1706\n",
      "100\t537\t550\n",
      "500\t901\t1734\n",
      "494\t117\t377\n",
      "Total number of sequences before pre.cluster was 494.\n",
      "pre.cluster removed 377 sequences.\n",
      "\n",
      "600\t886\t1749\n",
      "700\t881\t1754\n",
      "200\t486\t601\n",
      "800\t874\t1761\n",
      "900\t868\t1767\n",
      "1000\t867\t1768\n",
      "300\t470\t617\n",
      "1100\t862\t1773\n",
      "1200\t853\t1782\n",
      "1300\t852\t1783\n",
      "400\t459\t628\n",
      "1400\t851\t1784\n",
      "1500\t848\t1787\n",
      "1600\t846\t1789\n",
      "1700\t842\t1793\n",
      "500\t458\t629\n",
      "1800\t840\t1795\n",
      "1900\t834\t1801\n",
      "2000\t831\t1804\n",
      "2100\t828\t1807\n",
      "600\t458\t629\n",
      "2200\t828\t1807\n",
      "2300\t828\t1807\n",
      "700\t455\t632\n",
      "2400\t827\t1808\n",
      "2500\t827\t1808\n",
      "2600\t827\t1808\n",
      "800\t450\t637\n",
      "900\t447\t640\n",
      "1000\t446\t641\n",
      "0\t2613\t124\n",
      "2635\t827\t1808\n",
      "Total number of sequences before pre.cluster was 2635.\n",
      "pre.cluster removed 1808 sequences.\n",
      "\n",
      "It took 0 secs to cluster 494 sequences.\n",
      "1087\t446\t641\n",
      "Total number of sequences before pre.cluster was 1087.\n",
      "pre.cluster removed 641 sequences.\n",
      "\n",
      "100\t1288\t1449\n",
      "200\t1175\t1562\n",
      "It took 0 secs to cluster 2635 sequences.\n",
      "\n",
      "Processing group F3D3:\n",
      "300\t1091\t1646\n",
      "400\t1045\t1692\n",
      "500\t1010\t1727\n",
      "600\t993\t1744\n",
      "700\t985\t1752\n",
      "It took 0 secs to cluster 1087 sequences.\n",
      "\n",
      "Processing group F3D142:\n",
      "800\t975\t1762\n",
      "900\t967\t1770\n",
      "1000\t954\t1783\n",
      "1100\t946\t1791\n",
      "1200\t937\t1800\n",
      "1300\t934\t1803\n",
      "1400\t927\t1810\n",
      "0\t977\t72\n",
      "0\t686\t22\n",
      "1500\t924\t1813\n",
      "1600\t921\t1816\n",
      "1700\t917\t1820\n",
      "1800\t913\t1824\n",
      "1900\t909\t1828\n",
      "2000\t907\t1830\n",
      "100\t481\t568\n",
      "2100\t902\t1835\n",
      "2200\t900\t1837\n",
      "2300\t900\t1837\n",
      "2400\t900\t1837\n",
      "2500\t899\t1838\n",
      "200\t442\t607\n",
      "100\t387\t321\n",
      "2600\t898\t1839\n",
      "2700\t897\t1840\n",
      "300\t434\t615\n",
      "400\t427\t622\n",
      "500\t413\t636\n",
      "200\t354\t354\n",
      "600\t406\t643\n",
      "2737\t897\t1840\n",
      "Total number of sequences before pre.cluster was 2737.\n",
      "pre.cluster removed 1840 sequences.\n",
      "\n",
      "700\t402\t647\n",
      "800\t401\t648\n",
      "900\t401\t648\n",
      "1000\t401\t648\n",
      "300\t341\t367\n",
      "400\t332\t376\n",
      "500\t325\t383\n",
      "600\t322\t386\n",
      "700\t322\t386\n",
      "708\t322\t386\n",
      "Total number of sequences before pre.cluster was 708.\n",
      "pre.cluster removed 386 sequences.\n",
      "\n",
      "1049\t401\t648\n",
      "Total number of sequences before pre.cluster was 1049.\n",
      "pre.cluster removed 648 sequences.\n",
      "\n",
      "It took 0 secs to cluster 2737 sequences.\n",
      "\n",
      "Processing group F3D148:\n",
      "It took 0 secs to cluster 1049 sequences.\n",
      "\n",
      "Processing group F3D5:\n",
      "It took 0 secs to cluster 708 sequences.\n",
      "\n",
      "Processing group F3D143:\n",
      "0\t851\t32\n",
      "100\t474\t409\n",
      "0\t699\t19\n",
      "200\t433\t450\n",
      "300\t428\t455\n",
      "0\t1979\t108\n",
      "400\t425\t458\n",
      "500\t417\t466\n",
      "600\t413\t470\n",
      "700\t408\t475\n",
      "800\t407\t476\n",
      "100\t398\t320\n",
      "883\t407\t476\n",
      "Total number of sequences before pre.cluster was 883.\n",
      "pre.cluster removed 476 sequences.\n",
      "\n",
      "200\t370\t348\n",
      "300\t361\t357\n",
      "100\t989\t1098\n",
      "400\t351\t367\n",
      "500\t347\t371\n",
      "600\t342\t376\n",
      "700\t340\t378\n",
      "200\t868\t1219\n",
      "718\t340\t378\n",
      "Total number of sequences before pre.cluster was 718.\n",
      "pre.cluster removed 378 sequences.\n",
      "\n",
      "300\t809\t1278\n",
      "It took 0 secs to cluster 883 sequences.\n",
      "400\t787\t1300\n",
      "500\t773\t1314\n",
      "600\t761\t1326\n",
      "700\t751\t1336\n",
      "800\t748\t1339\n",
      "900\t738\t1349\n",
      "1000\t733\t1354\n",
      "1100\t729\t1358\n",
      "1200\t726\t1361\n",
      "1300\t724\t1363\n",
      "1400\t721\t1366\n",
      "1500\t717\t1370\n",
      "1600\t715\t1372\n",
      "1700\t714\t1373\n",
      "1800\t714\t1373\n",
      "1900\t713\t1374\n",
      "2000\t712\t1375\n",
      "2087\t712\t1375\n",
      "Total number of sequences before pre.cluster was 2087.\n",
      "pre.cluster removed 1375 sequences.\n",
      "\n",
      "It took 0 secs to cluster 718 sequences.\n",
      "It took 0 secs to cluster 2087 sequences.\n",
      "It took 1 secs to run pre.cluster.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.fasta\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.count_table\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D0.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D1.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D141.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D142.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D143.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D144.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D145.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D146.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D147.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D148.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D149.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D150.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D2.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D3.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D5.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D6.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D7.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D8.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.F3D9.map\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.Mock.map\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.pre.cluster(fasta='stability.trim.contigs.good.unique.filter.unique.fasta', count='stability.trim.contigs.good.unique.filter.count_table', diffs=2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > chimera.vsearch(fasta=stability.trim.contigs.good.unique.filter.unique.precluster.fasta,count=stability.trim.contigs.good.unique.filter.unique.precluster.count_table,dereplicate=t)\n",
      "\n",
      "Using 4 processors.\n",
      "C:\\Users\\campen\\mothur\\1.39.5\\mothur\\mothur\\vsearch.exe file does not exist. Checking path... \n",
      "Found vsearch in your path, using C:\\Users\\campen\\mothur\\1.39.5\\mothur\\vsearch.exe\n",
      "Checking sequences from C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.fasta ...\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "135989 nt in 558 seqs, min 242, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 261 (46.8%) chimeras, 290 (52.0%) non-chimeras,\n",
      "and 7 (1.3%) borderline sequences in 558 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "414 (6.2%) chimeras, 6217 (93.7%) non-chimeras,\n",
      "and 7 (0.1%) borderline sequences in 6638 total sequences.\n",
      "\n",
      "It took 0 secs to check 558 sequences from group F3D0.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "115802 nt in 475 seqs, min 242, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 216 (45.5%) chimeras, 254 (53.5%) non-chimeras,\n",
      "and 5 (1.1%) borderline sequences in 475 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "299 (6.0%) chimeras, 4666 (93.9%) non-chimeras,\n",
      "and 5 (0.1%) borderline sequences in 4970 total sequences.\n",
      "\n",
      "It took 1 secs to check 475 sequences from group F3D1.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "108661 nt in 446 seqs, min 240, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 220 (49.3%) chimeras, 223 (50.0%) non-chimeras,\n",
      "and 3 (0.7%) borderline sequences in 446 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "421 (8.3%) chimeras, 4651 (91.6%) non-chimeras,\n",
      "and 3 (0.1%) borderline sequences in 5075 total sequences.\n",
      "\n",
      "It took 0 secs to check 446 sequences from group F3D141.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "78463 nt in 322 seqs, min 243, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 128 (39.8%) chimeras, 191 (59.3%) non-chimeras,\n",
      "and 3 (0.9%) borderline sequences in 322 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "242 (9.1%) chimeras, 2423 (90.7%) non-chimeras,\n",
      "and 5 (0.2%) borderline sequences in 2670 total sequences.\n",
      "\n",
      "It took 0 secs to check 322 sequences from group F3D142.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "82850 nt in 340 seqs, min 242, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 156 (45.9%) chimeras, 182 (53.5%) non-chimeras,\n",
      "and 2 (0.6%) borderline sequences in 340 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "275 (10.3%) chimeras, 2395 (89.6%) non-chimeras,\n",
      "and 2 (0.1%) borderline sequences in 2672 total sequences.\n",
      "\n",
      "It took 1 secs to check 340 sequences from group F3D143.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "101826 nt in 418 seqs, min 240, max 248, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 174 (41.6%) chimeras, 240 (57.4%) non-chimeras,\n",
      "and 4 (1.0%) borderline sequences in 418 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "429 (11.1%) chimeras, 3441 (88.8%) non-chimeras,\n",
      "and 5 (0.1%) borderline sequences in 3875 total sequences.\n",
      "\n",
      "It took 0 secs to check 418 sequences from group F3D144.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "123975 nt in 509 seqs, min 242, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 221 (43.4%) chimeras, 285 (56.0%) non-chimeras,\n",
      "and 3 (0.6%) borderline sequences in 509 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "584 (9.6%) chimeras, 5527 (90.4%) non-chimeras,\n",
      "and 3 (0.0%) borderline sequences in 6114 total sequences.\n",
      "\n",
      "It took 0 secs to check 509 sequences from group F3D145.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "127472 nt in 523 seqs, min 242, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 249 (47.6%) chimeras, 272 (52.0%) non-chimeras,\n",
      "and 2 (0.4%) borderline sequences in 523 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "460 (10.7%) chimeras, 3839 (89.3%) non-chimeras,\n",
      "and 2 (0.0%) borderline sequences in 4301 total sequences.\n",
      "\n",
      "It took 1 secs to check 523 sequences from group F3D146.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "218496 nt in 897 seqs, min 240, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 472 (52.6%) chimeras, 417 (46.5%) non-chimeras,\n",
      "and 8 (0.9%) borderline sequences in 897 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "1688 (11.9%) chimeras, 12473 (88.0%) non-chimeras,\n",
      "and 9 (0.1%) borderline sequences in 14170 total sequences.\n",
      "\n",
      "It took 1 secs to check 897 sequences from group F3D147.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "173455 nt in 712 seqs, min 241, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 365 (51.3%) chimeras, 338 (47.5%) non-chimeras,\n",
      "and 9 (1.3%) borderline sequences in 712 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "1037 (9.9%) chimeras, 9470 (90.0%) non-chimeras,\n",
      "and 17 (0.2%) borderline sequences in 10524 total sequences.\n",
      "\n",
      "It took 0 secs to check 712 sequences from group F3D148.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "199108 nt in 817 seqs, min 240, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 456 (55.8%) chimeras, 356 (43.6%) non-chimeras,\n",
      "and 5 (0.6%) borderline sequences in 817 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "1034 (9.3%) chimeras, 10051 (90.6%) non-chimeras,\n",
      "and 6 (0.1%) borderline sequences in 11091 total sequences.\n",
      "\n",
      "It took 1 secs to check 817 sequences from group F3D149.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "129640 nt in 532 seqs, min 242, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 267 (50.2%) chimeras, 261 (49.1%) non-chimeras,\n",
      "and 4 (0.8%) borderline sequences in 532 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "475 (10.3%) chimeras, 4123 (89.5%) non-chimeras,\n",
      "and 7 (0.2%) borderline sequences in 4605 total sequences.\n",
      "\n",
      "It took 1 secs to check 532 sequences from group F3D150.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "201544 nt in 827 seqs, min 241, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 452 (54.7%) chimeras, 363 (43.9%) non-chimeras,\n",
      "and 12 (1.5%) borderline sequences in 827 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "1045 (6.2%) chimeras, 15731 (93.7%) non-chimeras,\n",
      "and 17 (0.1%) borderline sequences in 16793 total sequences.\n",
      "\n",
      "It took 0 secs to check 827 sequences from group F3D2.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "97700 nt in 401 seqs, min 242, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 183 (45.6%) chimeras, 214 (53.4%) non-chimeras,\n",
      "and 4 (1.0%) borderline sequences in 401 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "462 (8.1%) chimeras, 5207 (91.7%) non-chimeras,\n",
      "and 9 (0.2%) borderline sequences in 5678 total sequences.\n",
      "\n",
      "It took 1 secs to check 401 sequences from group F3D3.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "99208 nt in 407 seqs, min 242, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 173 (42.5%) chimeras, 231 (56.8%) non-chimeras,\n",
      "and 3 (0.7%) borderline sequences in 407 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "237 (6.4%) chimeras, 3476 (93.5%) non-chimeras,\n",
      "and 3 (0.1%) borderline sequences in 3716 total sequences.\n",
      "\n",
      "It took 0 secs to check 407 sequences from group F3D5.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "120137 nt in 493 seqs, min 241, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 227 (46.0%) chimeras, 263 (53.3%) non-chimeras,\n",
      "and 3 (0.6%) borderline sequences in 493 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "405 (5.9%) chimeras, 6430 (94.0%) non-chimeras,\n",
      "and 4 (0.1%) borderline sequences in 6839 total sequences.\n",
      "\n",
      "It took 1 secs to check 493 sequences from group F3D6.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "80171 nt in 329 seqs, min 242, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 130 (39.5%) chimeras, 198 (60.2%) non-chimeras,\n",
      "and 1 (0.3%) borderline sequences in 329 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "290 (6.6%) chimeras, 4073 (93.3%) non-chimeras,\n",
      "and 2 (0.0%) borderline sequences in 4365 total sequences.\n",
      "\n",
      "It took 0 secs to check 329 sequences from group F3D7.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "102382 nt in 420 seqs, min 241, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 160 (38.1%) chimeras, 254 (60.5%) non-chimeras,\n",
      "and 6 (1.4%) borderline sequences in 420 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "227 (5.0%) chimeras, 4287 (94.8%) non-chimeras,\n",
      "and 10 (0.2%) borderline sequences in 4524 total sequences.\n",
      "\n",
      "It took 0 secs to check 420 sequences from group F3D8.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "126510 nt in 519 seqs, min 242, max 246, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 235 (45.3%) chimeras, 278 (53.6%) non-chimeras,\n",
      "and 6 (1.2%) borderline sequences in 519 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "346 (5.7%) chimeras, 5765 (94.2%) non-chimeras,\n",
      "and 8 (0.1%) borderline sequences in 6119 total sequences.\n",
      "\n",
      "It took 1 secs to check 519 sequences from group F3D9.\n",
      "vsearch v2.3.4_win_x86_64, 63.9GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.temp 100%\n",
      "28538 nt in 117 seqs, min 243, max 248, avg 244\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting unique k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 62 (53.0%) chimeras, 55 (47.0%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 117 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "78 (1.9%) chimeras, 4055 (98.1%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 4133 total sequences.\n",
      "\n",
      "It took 0 secs to check 117 sequences from group Mock.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.count_table\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.chimeras\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.accnos\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.chimera.vsearch(fasta='stability.trim.contigs.good.unique.filter.unique.precluster.fasta', count='stability.trim.contigs.good.unique.filter.unique.precluster.count_table', dereplicate='t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > remove.seqs(fasta=stability.trim.contigs.good.unique.filter.unique.precluster.fasta,accnos=stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.accnos)\n",
      "[WARNING]: This command can take a namefile and you did not provide one. The current namefile is C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.names which seems to match C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.fasta.\n",
      "Removed 3350 sequences from your fasta file.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.fasta\n",
      "\n",
      "\n",
      "<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<^>>>>>>>>>>>>>>>>>>>>>>>>>>>>>\n",
      "<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<^>>>>>>>>>>>>>>>>>>>>>>>>>>>>>\n",
      "<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<^>>>>>>>>>>>>>>>>>>>>>>>>>>>>>\n",
      "Detected 1 [WARNING] messages, please review.\n",
      "<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<^>>>>>>>>>>>>>>>>>>>>>>>>>>>>>\n",
      "<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<^>>>>>>>>>>>>>>>>>>>>>>>>>>>>>\n",
      "<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<^>>>>>>>>>>>>>>>>>>>>>>>>>>>>>\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.remove.seqs(fasta='stability.trim.contigs.good.unique.filter.unique.precluster.fasta', accnos='stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.accnos')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > summary.seqs(fasta=current,count=current)\n",
      "Using C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.count_table as input file for the count parameter.\n",
      "Using C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.fasta as input file for the fasta parameter.\n",
      "\n",
      "Using 4 processors.\n",
      "\n",
      "\t\tStart\tEnd\tNBases\tAmbigs\tPolymer\tNumSeqs\n",
      "Minimum:\t1\t369\t240\t0\t3\t1\n",
      "2.5%-tile:\t1\t369\t243\t0\t3\t2961\n",
      "25%-tile:\t1\t369\t243\t0\t4\t29607\n",
      "Median: \t1\t369\t243\t0\t4\t59213\n",
      "75%-tile:\t1\t369\t244\t0\t5\t88819\n",
      "97.5%-tile:\t1\t369\t244\t0\t6\t115464\n",
      "Maximum:\t3\t369\t248\t0\t7\t118424\n",
      "Mean:\t1.00003\t369\t243.464\t0\t4.37488\n",
      "# of unique seqs:\t2283\n",
      "total # of seqs:\t118424\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.summary\n",
      "\n",
      "It took 0 secs to summarize 118424 sequences.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.summary.seqs(fasta='current', count='current')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > classify.seqs(fasta=stability.trim.contigs.good.unique.filter.unique.precluster.pick.fasta,count=stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.count_table,reference=trainset9_032012.pds.fasta,taxonomy=trainset9_032012.pds.tax,cutoff=80)\n",
      "\n",
      "Using 4 processors.\n",
      "Generating search database...    DONE.\n",
      "It took 4 seconds generate search database. \n",
      "\n",
      "Reading in the C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\trainset9_032012.pds.tax taxonomy...\tDONE.\n",
      "Calculating template taxonomy tree...     DONE.\n",
      "Calculating template probabilities...     DONE.\n",
      "It took 15 seconds get probabilities. \n",
      "Classifying sequences from C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.fasta ...\n",
      "Reading template taxonomy...     Reading template taxonomy...     Reading template taxonomy...     DONE.\n",
      "Reading template probabilities...     DONE.\n",
      "DONE.\n",
      "Reading template probabilities...     Reading template probabilities...     Processing sequence: 100\n",
      "Processing sequence: 200\n",
      "Processing sequence: 300\n",
      "Processing sequence: 400\n",
      "Processing sequence: 500\n",
      "Processing sequence: 573\n",
      "DONE.\n",
      "It took 24 seconds get probabilities. \n",
      "DONE.\n",
      "It took 24 seconds get probabilities. \n",
      "DONE.\n",
      "It took 25 seconds get probabilities. \n",
      "Processing sequence: 100\n",
      "Processing sequence: 100\n",
      "Processing sequence: 100\n",
      "Processing sequence: 200\n",
      "Processing sequence: 200\n",
      "Processing sequence: 200\n",
      "Processing sequence: 300\n",
      "Processing sequence: 300\n",
      "Processing sequence: 300\n",
      "Processing sequence: 400\n",
      "Processing sequence: 400\n",
      "Processing sequence: 400\n",
      "Processing sequence: 500\n",
      "Processing sequence: 500\n",
      "Processing sequence: 500\n",
      "Processing sequence: 570\n",
      "Processing sequence: 570\n",
      "Processing sequence: 570\n",
      "\n",
      "It took 28 secs to classify 2283 sequences.\n",
      "\n",
      "\n",
      "It took 0 secs to create the summary file for 2283 sequences.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pds.wang.taxonomy\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pds.wang.tax.summary\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.classify.seqs(fasta='stability.trim.contigs.good.unique.filter.unique.precluster.pick.fasta', count='stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.count_table', reference='trainset9_032012.pds.fasta', taxonomy='trainset9_032012.pds.tax', cutoff=80)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > remove.lineage(fasta=stability.trim.contigs.good.unique.filter.unique.precluster.pick.fasta,count=stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.count_table,taxonomy=stability.trim.contigs.good.unique.filter.unique.precluster.pick.pds.wang.taxonomy,taxon=Chloroplast-Mitochondria-unknown-Archaea-Eukaryota)\n",
      "\n",
      "[NOTE]: The count file should contain only unique names, so mothur assumes your fasta, list and taxonomy files also contain only uniques.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pds.wang.pick.taxonomy\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.fasta\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.pick.count_table\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.remove.lineage(fasta='stability.trim.contigs.good.unique.filter.unique.precluster.pick.fasta', count='stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.count_table', taxonomy='stability.trim.contigs.good.unique.filter.unique.precluster.pick.pds.wang.taxonomy', taxon='Chloroplast-Mitochondria-unknown-Archaea-Eukaryota')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > summary.tax(taxonomy=current,count=current)\n",
      "Using C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.pick.count_table as input file for the count parameter.\n",
      "Using C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pds.wang.pick.taxonomy as input file for the taxonomy parameter.\n",
      "\n",
      "It took 0 secs to create the summary file for 118254 sequences.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pds.wang.pick.tax.summary\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.summary.tax(taxonomy='current', count='current')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# mothur = mothur.get.groups(count='stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.pick.count_table', fasta='stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.fasta', groups='Mock')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# mothur = mothur.seq.error(fasta='stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.fasta', count='stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table', reference='MiSeq_SOP\\HMP_MOCK.v35.fasta', aligned='F')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# mothur = mothur.dist.seqs(fasta='stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.fasta', cutoff=0.03)\n",
    "# mothur = mothur.cluster(column='stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.dist', count='stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table')\n",
    "# mothur = mothur.make.shared(list='stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.opti_mcc.list', count='stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table', label=0.03)\n",
    "# mothur = mothur.rarefaction.single(shared='stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.opti_mcc.shared')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > remove.groups(count=stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.pick.count_table,fasta=stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.fasta,taxonomy=stability.trim.contigs.good.unique.filter.unique.precluster.pick.pds.wang.pick.taxonomy,groups=Mock)\n",
      "\n",
      "[NOTE]: The count file should contain only unique names, so mothur assumes your fasta, list and taxonomy files also contain only uniques.\n",
      "\n",
      "Removed 32 sequences from your fasta file.\n",
      "Removed 4055 sequences from your count file.\n",
      "Removed 32 sequences from your taxonomy file.\n",
      "\n",
      "Output File names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.fasta\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pds.wang.pick.pick.taxonomy\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.remove.groups(count='stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.pick.count_table', fasta='stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.fasta', taxonomy='stability.trim.contigs.good.unique.filter.unique.precluster.pick.pds.wang.pick.taxonomy', groups='Mock')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > dist.seqs(fasta=stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.fasta,cutoff=0.03)\n",
      "\n",
      "Using 4 processors.\n",
      "0\t0\n",
      "100\t0\n",
      "200\t0\n",
      "1600\t1\n",
      "300\t1\n",
      "400\t1\n",
      "1200\t1\n",
      "2000\t1\n",
      "500\t1\n",
      "600\t1\n",
      "1700\t1\n",
      "1300\t1\n",
      "700\t1\n",
      "2100\t1\n",
      "800\t1\n",
      "1400\t1\n",
      "1800\t1\n",
      "900\t1\n",
      "1500\t1\n",
      "1000\t1\n",
      "1900\t1\n",
      "2200\t1\n",
      "1100\t2\n",
      "354\t2\n",
      "299\t2\n",
      "1113\t2\n",
      "462\t2\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.dist\n",
      "\n",
      "It took 2 seconds to calculate the distances for 2229 sequences.\n",
      "\n",
      "mothur > cluster(column=stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.dist,count=stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table)\n",
      "\n",
      "Using 4 processors.\n",
      "\n",
      "You did not set a cutoff, using 0.03.\n",
      "\n",
      "Clustering C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.dist\n",
      "\n",
      "\n",
      "iter\ttime\tlabel\tnum_otus\tcutoff\ttp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "0\t0\t0.03\t2229\t0.03\t0\t2463014\t0\t20092\t0\t1\t0\t0.991909\t0\t0.991909\t0\t0\t\n",
      "1\t0\t0.03\t526\t0.03\t16813\t2461389\t1625\t3279\t0.836801\t0.99934\t0.911867\t0.99867\t0.0881332\t0.998025\t0.872546\t0.872723\t\n",
      "2\t0\t0.03\t490\t0.03\t17193\t2461360\t1654\t2899\t0.855714\t0.999328\t0.912241\t0.998824\t0.0877593\t0.998166\t0.882609\t0.883074\t\n",
      "3\t0\t0.03\t487\t0.03\t17208\t2461349\t1665\t2884\t0.85646\t0.999324\t0.911779\t0.99883\t0.0882213\t0.998168\t0.882771\t0.883254\t\n",
      "4\t0\t0.03\t488\t0.03\t17204\t2461354\t1660\t2888\t0.856261\t0.999326\t0.912002\t0.998828\t0.0879983\t0.998168\t0.882776\t0.883253\t\n",
      "\n",
      "\n",
      "It took 0 seconds to cluster\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.opti_mcc.list\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.opti_mcc.steps\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.opti_mcc.sensspec\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.dist.seqs(fasta='stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.fasta', cutoff=0.03)\n",
    "mothur = mothur.cluster(column='stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.dist', count='stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > make.shared(list=stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.opti_mcc.list,count=stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table,label=0.03)\n",
      "0.03\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.opti_mcc.shared\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.make.shared(list='stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.opti_mcc.list', count='stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table', label=0.03)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > classify.otu(list=stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.opti_mcc.list,count=stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table,taxonomy=stability.trim.contigs.good.unique.filter.unique.precluster.pick.pds.wang.pick.pick.taxonomy,label=0.03)\n",
      "0.03\t488\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.opti_mcc.0.03.cons.taxonomy\n",
      "C:\\Users\\campen\\Documents\\coding\\python\\rhea\\example\\Mothur_MiSeq_SOP\\stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.opti_mcc.0.03.cons.tax.summary\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mothur = mothur.classify.otu(list='stability.trim.contigs.good.unique.filter.unique.precluster.pick.pick.pick.opti_mcc.list', count='stability.trim.contigs.good.unique.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table', taxonomy='stability.trim.contigs.good.unique.filter.unique.precluster.pick.pds.wang.pick.pick.taxonomy', label=0.03)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'rhea.Mothur'"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pprint import pprint\n",
    "mothur"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
